diff --git a/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux_x86_64.cpp b/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux_x86_64.cpp
index c6aa320..e222909 100644
--- a/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux_x86_64.cpp
+++ b/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux_x86_64.cpp
@@ -1040,14 +1042,16 @@ NativeRegisterContextLinux_x86_64::GetSyscallData() {
     static const uint32_t Args[] = {lldb_eax_i386, lldb_ebx_i386, lldb_ecx_i386,
                                     lldb_edx_i386, lldb_esi_i386, lldb_edi_i386,
                                     lldb_ebp_i386};
-    return SyscallData{Int80, Args, lldb_eax_i386};
+    return SyscallData{llvm::makeArrayRef(Int80), llvm::makeArrayRef(Args),
+                       lldb_eax_i386};
   }
   case llvm::Triple::x86_64: {
     static const uint8_t Syscall[] = {0x0f, 0x05};
     static const uint32_t Args[] = {
         lldb_rax_x86_64, lldb_rdi_x86_64, lldb_rsi_x86_64, lldb_rdx_x86_64,
         lldb_r10_x86_64, lldb_r8_x86_64,  lldb_r9_x86_64};
-    return SyscallData{Syscall, Args, lldb_rax_x86_64};
+    return SyscallData{llvm::makeArrayRef(Syscall), llvm::makeArrayRef(Args),
+                       lldb_rax_x86_64};
   }
   default:
     llvm_unreachable("Unhandled architecture!");
