diff --git a/plugins/python/uwsgiplugin.py b/plugins/python/uwsgiplugin.py
index f9b4ddb..ca99479 100644
--- a/plugins/python/uwsgiplugin.py
+++ b/plugins/python/uwsgiplugin.py
@@ -45,6 +45,8 @@ if not 'UWSGI_PYTHON_NOLIB' in os.environ:
                 libpath = '%s/%s' % (libdir, sysconfig.get_config_var('LIBRARY'))
         if not os.path.exists(libpath): 
             libpath = '%s/libpython%s.a' % (libdir, version)
+        if not os.path.exists(libpath):
+            libpath = '%s/libpython%s.tbd' % (libdir, version)
         LIBS.append(libpath)
         # hack for messy linkers/compilers
         if '-lutil' in LIBS:
