diff --git a/fceux-server/server.cpp b/fceux-server/server.cpp
index 116e8f4..6ac11ef 100644
--- a/fceux-server/server.cpp
+++ b/fceux-server/server.cpp
@@ -114,7 +114,7 @@ int LoadConfigFile(char *fn)
  if(fp=fopen(fn,"rb"))
  {
   char buf[256];
-  while(fgets(buf, 256, fp) > 0)
+  while(fgets(buf, 256, fp))
   {
    if(!strncasecmp(buf,"maxclients",strlen("maxclients")))
     sscanf(buf,"%*s %d",&ServerConfig.MaxClients);
diff --git a/src/cheat.cpp b/src/cheat.cpp
index f61668a..fa5ae8d 100644
--- a/src/cheat.cpp
+++ b/src/cheat.cpp
@@ -216,7 +216,7 @@ void FCEU_LoadGameCheats(FILE *override)
 	}
 
 	FCEU_DispMessage("Cheats file loaded.",0); //Tells user a cheats file was loaded.
-	while(fgets(linebuf,2048,fp)>0)
+	while(fgets(linebuf,2048,fp))
 	{
 		char *tbuf=linebuf;
 		int doc=0;
