diff -pur a/src/lib/util/harddisk.cpp b/src/lib/util/harddisk.cpp
--- a/src/lib/util/harddisk.cpp	2017-08-30 01:05:26.000000000 +0100
+++ b/src/lib/util/harddisk.cpp	2017-09-06 23:09:11.000000000 +0100
@@ -19,12 +19,6 @@
     TYPE DEFINITIONS
 ***************************************************************************/
 
-struct hard_disk_file
-{
-	chd_file *          chd;                /* CHD file */
-	hard_disk_info      info;               /* hard disk info */
-};
-
 
 
 /***************************************************************************
diff -pur a/src/lib/util/harddisk.h b/src/lib/util/harddisk.h
--- a/src/lib/util/harddisk.h	2017-08-30 01:05:26.000000000 +0100
+++ b/src/lib/util/harddisk.h	2017-09-06 23:09:19.000000000 +0100
@@ -21,8 +21,6 @@
     TYPE DEFINITIONS
 ***************************************************************************/
 
-struct hard_disk_file;
-
 struct hard_disk_info
 {
 	uint32_t          cylinders;
@@ -31,6 +29,12 @@ struct hard_disk_info
 	uint32_t          sectorbytes;
 };
 
+struct hard_disk_file
+{
+	chd_file *          chd;                /* CHD file */
+	hard_disk_info      info;               /* hard disk info */
+};
+
 
 
 /***************************************************************************
