diff --git i/Makefile.in w/Makefile.in
index 9ea86fa..2e384af 100644
--- i/Makefile.in
+++ w/Makefile.in
@@ -30415,7 +30415,7 @@ src-maintainer-clean: src-distclean
 $(scripts_java_JAVA_CLASSES) : %.class : %.java | scripts/java/$(octave_dirstamp)
 	$(OCT_V_JAVAC)$(MKDIR_P) scripts/java/$(org_octave_dir) && \
 	( cd $(srcdir)/scripts/java; \
-	  "$(JAVAC)" -source 1.6 -target 1.6 -Xlint:-options \
+	  "$(JAVAC)" -source 1.7 -target 1.7 -Xlint:-options \
 	             -d $(abs_top_builddir)/scripts/java \
 	             $(org_octave_dir)/$(<F) )
 
diff --git i/scripts/java/module.mk w/scripts/java/module.mk
index 2923049..9af8e94 100644
--- i/scripts/java/module.mk
+++ w/scripts/java/module.mk
@@ -47,7 +47,7 @@ oct__v_JAVAC_1 =
 $(%canon_reldir%_JAVA_CLASSES) : %.class : %.java | %reldir%/$(octave_dirstamp)
 	$(OCT_V_JAVAC)$(MKDIR_P) %reldir%/$(org_octave_dir) && \
 	( cd $(srcdir)/scripts/java; \
-	  "$(JAVAC)" -source 1.6 -target 1.6 -Xlint:-options \
+	  "$(JAVAC)" -source 1.7 -target 1.7 -Xlint:-options \
 	             -d $(abs_top_builddir)/scripts/java \
 	             $(org_octave_dir)/$(<F) )
 
