diff --git a/qtwebengine/src/3rdparty/chromium/printing/backend/print_backend_cups.cc b/qtwebengine/src/3rdparty/chromium/printing/backend/print_backend_cups.cc
index 2b1033e..d76e1de 100644
--- a/qtwebengine/src/3rdparty/chromium/printing/backend/print_backend_cups.cc
+++ b/qtwebengine/src/3rdparty/chromium/printing/backend/print_backend_cups.cc
@@ -4,6 +4,7 @@
 
 #include "printing/backend/print_backend.h"
 
+#include <cups/ppd.h>
 #include <dlfcn.h>
 #include <errno.h>
 #include <pthread.h>
