diff --git a/qtwebengine/src/3rdparty/chromium/chrome/installer/mac/third_party/bsdiff/goobsdiff.gyp b/qtwebengine/src/3rdparty/chromium/chrome/installer/mac/third_party/bsdiff/goobsdiff.gyp
index 68e4668..bde3b1c 100644
--- a/qtwebengine/src/3rdparty/chromium/chrome/installer/mac/third_party/bsdiff/goobsdiff.gyp
+++ b/qtwebengine/src/3rdparty/chromium/chrome/installer/mac/third_party/bsdiff/goobsdiff.gyp
@@ -9,8 +9,8 @@
     ],
     'link_settings': {
       'libraries': [
-        '$(SDKROOT)/usr/lib/libbz2.dylib',
-        '$(SDKROOT)/usr/lib/libz.dylib',
+        '/usr/lib/libbz2.dylib',
+        '/usr/lib/libz.dylib',
       ],
     },
     'configurations': {
diff --git a/qtwebengine/src/3rdparty/chromium/content/content_browser.gypi b/qtwebengine/src/3rdparty/chromium/content/content_browser.gypi
index cb22dc5..fd0d52c 100644
--- a/qtwebengine/src/3rdparty/chromium/content/content_browser.gypi
+++ b/qtwebengine/src/3rdparty/chromium/content/content_browser.gypi
@@ -1849,7 +1849,7 @@
       ],
       'link_settings': {
         'libraries': [
-          '$(SDKROOT)/usr/lib/libbsm.dylib',
+          '/usr/lib/libbsm.dylib',
         ],
       },
     }],
diff --git a/qtwebengine/src/3rdparty/chromium/net/net.gyp b/qtwebengine/src/3rdparty/chromium/net/net.gyp
index fa0f6f8..9036cf3 100644
--- a/qtwebengine/src/3rdparty/chromium/net/net.gyp
+++ b/qtwebengine/src/3rdparty/chromium/net/net.gyp
@@ -427,7 +427,7 @@
                 '$(SDKROOT)/System/Library/Frameworks/Foundation.framework',
                 '$(SDKROOT)/System/Library/Frameworks/Security.framework',
                 '$(SDKROOT)/System/Library/Frameworks/SystemConfiguration.framework',
-                '$(SDKROOT)/usr/lib/libresolv.dylib',
+                '/usr/lib/libresolv.dylib',
               ]
             },
           },
@@ -442,7 +442,7 @@
                 '$(SDKROOT)/System/Library/Frameworks/MobileCoreServices.framework',
                 '$(SDKROOT)/System/Library/Frameworks/Security.framework',
                 '$(SDKROOT)/System/Library/Frameworks/SystemConfiguration.framework',
-                '$(SDKROOT)/usr/lib/libresolv.dylib',
+                '/usr/lib/libresolv.dylib',
               ],
             },
           },
diff --git a/qtwebengine/src/3rdparty/chromium/printing/printing.gyp b/qtwebengine/src/3rdparty/chromium/printing/printing.gyp
index 31c0f1d..0d8789b 100644
--- a/qtwebengine/src/3rdparty/chromium/printing/printing.gyp
+++ b/qtwebengine/src/3rdparty/chromium/printing/printing.gyp
@@ -267,7 +267,7 @@
               ['OS=="mac"', {
                 'link_settings': {
                   'libraries': [
-                    '$(SDKROOT)/usr/lib/libcups.dylib',
+                    '/usr/lib/libcups.dylib',
                   ]
                 },
               }, {
diff --git a/qtwebengine/src/3rdparty/chromium/sandbox/mac/sandbox_mac.gypi b/qtwebengine/src/3rdparty/chromium/sandbox/mac/sandbox_mac.gypi
index a2a616f..d7c1085 100644
--- a/qtwebengine/src/3rdparty/chromium/sandbox/mac/sandbox_mac.gypi
+++ b/qtwebengine/src/3rdparty/chromium/sandbox/mac/sandbox_mac.gypi
@@ -38,7 +38,7 @@
       ],
       'link_settings': {
         'libraries': [
-          '$(SDKROOT)/usr/lib/libbsm.dylib',
+          '/usr/lib/libbsm.dylib',
         ],
       },
       'conditions': [
diff --git a/qtwebengine/src/3rdparty/chromium/third_party/ffmpeg/ffmpeg.gyp b/qtwebengine/src/3rdparty/chromium/third_party/ffmpeg/ffmpeg.gyp
index ba13a16..48b2a4e 100644
--- a/qtwebengine/src/3rdparty/chromium/third_party/ffmpeg/ffmpeg.gyp
+++ b/qtwebengine/src/3rdparty/chromium/third_party/ffmpeg/ffmpeg.gyp
@@ -370,7 +370,7 @@
               ],
               'link_settings': {
                 'libraries': [
-                  '$(SDKROOT)/usr/lib/libz.dylib',
+                  '/usr/lib/libz.dylib',
                 ],
               },
               'xcode_settings': {
diff --git a/qtwebengine/src/3rdparty/chromium/third_party/libxml/libxml.gyp b/qtwebengine/src/3rdparty/chromium/third_party/libxml/libxml.gyp
index b957a02..e4f0dcd 100644
--- a/qtwebengine/src/3rdparty/chromium/third_party/libxml/libxml.gyp
+++ b/qtwebengine/src/3rdparty/chromium/third_party/libxml/libxml.gyp
@@ -69,7 +69,7 @@
               },
               'link_settings': {
                 'libraries': [
-                  '$(SDKROOT)/usr/lib/libxml2.dylib',
+                  '/usr/lib/libxml2.dylib',
                 ],
               },
             }],
diff --git a/qtwebengine/src/3rdparty/chromium/third_party/skia/gyp/zlib.gyp b/qtwebengine/src/3rdparty/chromium/third_party/skia/gyp/zlib.gyp
index 78adc34..285d4d4 100644
--- a/qtwebengine/src/3rdparty/chromium/third_party/skia/gyp/zlib.gyp
+++ b/qtwebengine/src/3rdparty/chromium/third_party/skia/gyp/zlib.gyp
@@ -60,7 +60,7 @@
                 'conditions': [
                   [ 'skia_os == "mac" or skia_os == "ios"', {
                     'libraries': [
-                      '$(SDKROOT)/usr/lib/libz.dylib',
+                      '/usr/lib/libz.dylib',
                     ]
                   }, {  # skia_os != "mac" and skia_os != "ios"
                     'libraries': [
diff --git a/qtwebengine/src/3rdparty/chromium/third_party/sqlite/sqlite.gyp b/qtwebengine/src/3rdparty/chromium/third_party/sqlite/sqlite.gyp
index 2e48512..0acf0d0 100644
--- a/qtwebengine/src/3rdparty/chromium/third_party/sqlite/sqlite.gyp
+++ b/qtwebengine/src/3rdparty/chromium/third_party/sqlite/sqlite.gyp
@@ -48,7 +48,7 @@
               ],
               'link_settings': {
                 'libraries': [
-                  '$(SDKROOT)/usr/lib/libsqlite3.dylib',
+                  '/usr/lib/libsqlite3.dylib',
                 ],
               },
             }],
diff --git a/qtwebengine/src/3rdparty/chromium/tools/gyp/pylib/gyp/generator/xcode.py b/qtwebengine/src/3rdparty/chromium/tools/gyp/pylib/gyp/generator/xcode.py
index 907e048..00180eb 100644
--- a/qtwebengine/src/3rdparty/chromium/tools/gyp/pylib/gyp/generator/xcode.py
+++ b/qtwebengine/src/3rdparty/chromium/tools/gyp/pylib/gyp/generator/xcode.py
@@ -88,7 +88,7 @@ generator_extra_sources_for_rules = [
 # Xcode's standard set of library directories, which don't need to be duplicated
 # in LIBRARY_SEARCH_PATHS. This list is not exhaustive, but that's okay.
 xcode_standard_library_dirs = frozenset([
-  '$(SDKROOT)/usr/lib',
+  '/usr/lib',
   '$(SDKROOT)/usr/local/lib',
 ])