diff --git a/chromium/third_party/blink/renderer/core/editing/finder/text_finder.cc b/chromium/third_party/blink/renderer/core/editing/finder/text_finder.cc
index e41a894fc28..85df6865cfe 100644
--- a/chromium/third_party/blink/renderer/core/editing/finder/text_finder.cc
+++ b/chromium/third_party/blink/renderer/core/editing/finder/text_finder.cc
@@ -629,7 +629,8 @@ gfx::RectF TextFinder::ActiveFindMatchRect() {
   if (!current_active_match_frame_ || !active_match_)
     return gfx::RectF();
 
-  return gfx::RectF(FindInPageRectFromRange(EphemeralRange(ActiveMatch())));
+  FloatRect r = FindInPageRectFromRange(EphemeralRange(ActiveMatch()));
+  return gfx::RectF(r.X(), r.Y(), r.Width(), r.Height());
 }
 
 Vector<gfx::RectF> TextFinder::FindMatchRects() {
@@ -639,7 +640,8 @@ Vector<gfx::RectF> TextFinder::FindMatchRects() {
   match_rects.ReserveCapacity(match_rects.size() + find_matches_cache_.size());
   for (const FindMatch& match : find_matches_cache_) {
     DCHECK(!match.rect_.IsEmpty());
-    match_rects.push_back(match.rect_);
+    FloatRect r = match.rect_;
+    match_rects.push_back(gfx::RectF(r.X(), r.Y(), r.Width(), r.Height()));
   }
 
   return match_rects;
diff --git a/chromium/third_party/blink/renderer/core/paint/paint_timing_detector.cc b/chromium/third_party/blink/renderer/core/paint/paint_timing_detector.cc
index 48d06120e0c..df357910406 100644
--- a/chromium/third_party/blink/renderer/core/paint/paint_timing_detector.cc
+++ b/chromium/third_party/blink/renderer/core/paint/paint_timing_detector.cc
@@ -320,7 +320,7 @@ FloatRect PaintTimingDetector::BlinkSpaceToDIPs(
   // May be nullptr in tests.
   if (!widget)
     return float_rect;
-  return FloatRect(widget->BlinkSpaceToDIPs(gfx::RectF(float_rect)));
+  return FloatRect(widget->BlinkSpaceToDIPs(gfx::RectF(float_rect.X(), float_rect.Y(), float_rect.Width(), float_rect.Height())));
 }
 
 FloatRect PaintTimingDetector::CalculateVisualRect(
diff --git a/chromium/third_party/blink/renderer/modules/exported/web_ax_object.cc b/chromium/third_party/blink/renderer/modules/exported/web_ax_object.cc
index f7a348c812e..06cd41f00be 100644
--- a/chromium/third_party/blink/renderer/modules/exported/web_ax_object.cc
+++ b/chromium/third_party/blink/renderer/modules/exported/web_ax_object.cc
@@ -1480,7 +1480,7 @@ void WebAXObject::GetRelativeBounds(WebAXObject& offset_container,
   private_->GetRelativeBounds(&container, bounds, container_transform,
                               clips_children);
   offset_container = WebAXObject(container);
-  bounds_in_container = gfx::RectF(bounds);
+  bounds_in_container = gfx::RectF(bounds.X(), bounds.Y(), bounds.Width(), bounds.Height());
 }
 
 void WebAXObject::GetAllObjectsWithChangedBounds(
