diff --git a/qtwebengine/src/3rdparty/chromium/base/mac/sdk_forward_declarations.h b/qtwebengine/src/3rdparty/chromium/base/mac/sdk_forward_declarations.h
index 769d762..3569e04 100644
--- a/qtwebengine/src/3rdparty/chromium/base/mac/sdk_forward_declarations.h
+++ b/qtwebengine/src/3rdparty/chromium/base/mac/sdk_forward_declarations.h
@@ -15,6 +15,7 @@
 #import <CoreWLAN/CoreWLAN.h>
 #import <ImageCaptureCore/ImageCaptureCore.h>
 #import <IOBluetooth/IOBluetooth.h>
+#import <CoreBluetooth/CoreBluetooth.h>
 #include <stdint.h>
 
 #include "base/base_export.h"
