diff --git a/create.cc b/create.cc
index 11c4041..06d628c 100644
--- a/create.cc
+++ b/create.cc
@@ -26,6 +26,8 @@
 #if !defined __FreeBSD__ && !defined __OpenBSD__ && !defined __NetBSD__ && \
     !defined __DragonFly__ && !defined __APPLE__ && !defined __OS2__
 #include <sys/sysmacros.h>		// for major, minor
+#else
+#include <sys/types.h>	// for major, minor
 #endif
 #include <ftw.h>
 #include <grp.h>
diff --git a/decode.cc b/decode.cc
index 1742df2..6743491 100644
--- a/decode.cc
+++ b/decode.cc
@@ -29,6 +29,8 @@
 #if !defined __FreeBSD__ && !defined __OpenBSD__ && !defined __NetBSD__ && \
     !defined __DragonFly__ && !defined __APPLE__ && !defined __OS2__
 #include <sys/sysmacros.h>		// for major, minor, makedev
+#else
+#include <sys/types.h>	// for major, minor, makedev
 #endif
 #include <lzlib.h>
 
diff --git a/decode_lz.cc b/decode_lz.cc
index 15ac2a7..a31384b 100644
--- a/decode_lz.cc
+++ b/decode_lz.cc
@@ -29,6 +29,8 @@
 #if !defined __FreeBSD__ && !defined __OpenBSD__ && !defined __NetBSD__ && \
     !defined __DragonFly__ && !defined __APPLE__ && !defined __OS2__
 #include <sys/sysmacros.h>		// for major, minor, makedev
+#else
+#include <sys/types.h>	// for major, minor, makedev
 #endif
 #include <lzlib.h>
 
