diff --git a/pyproject.toml b/pyproject.toml
index 9e0990f..d89bb14 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,9 +1,11 @@
-[tool.poetry]
+[project]
 name = "pySigma"
 version = "0.11.19"
 license = "LGPL-2.1-only"
 description = "Sigma rule processing and conversion tools"
-authors = ["Thomas Patzke <thomas@patzke.org>"]
+authors = [
+  { name = "Thomas Patzke", email = "dsun20@bloomberg.net" },
+]
 readme = "README.md"
 repository = "https://github.com/SigmaHQ/pySigma"
 classifiers = [
@@ -13,12 +15,9 @@ classifiers = [
     "Programming Language :: Python :: 3.9",
     "Programming Language :: Python :: 3.10",
     "Programming Language :: Python :: 3.11",
-    "Programming Language :: Python :: 3.12",    
+    "Programming Language :: Python :: 3.12",
     "Topic :: Security"
 ]
-packages = [
-    { include = "sigma" }
-]
 
 [tool.poetry.dependencies]
 python = "^3.9"
@@ -28,6 +27,18 @@ pyyaml = "^6.0"
 requests = "^2.31"
 jinja2 = "^3.1"
 
+[project.urls]
+Repository = "https://github.com/SigmaHQ/pySigma"
+
+[build-system]
+requires = ["poetry-core>=1.8.1"]
+build-backend = "poetry.core.masonry.api"
+
+[tool.poetry]
+packages = [
+    { include = "sigma" }
+]
+
 [tool.poetry.group.dev.dependencies]
 black = "^24.4.2"
 mypy = "^1.8"
@@ -43,9 +54,5 @@ defusedxml = "^0.7"
 [tool.black]
 line-length = 100
 
-[build-system]
-requires = ["poetry-core>=1.8.1"]
-build-backend = "poetry.core.masonry.api"
-
 [tool.pytest.ini_options]
 filterwarnings = ["ignore::DeprecationWarning"]
