diff --git a/daemon/daemon.mk b/daemon/daemon.mk
index 12ccc63..516cb40 100644
--- a/daemon/daemon.mk
+++ b/daemon/daemon.mk
@@ -32,6 +32,7 @@ kresd_CFLAGS := -fPIE \
 		-Dlibknot_SONAME=\"$(libknot_SONAME)\" \
 		-Dlibzscanner_SONAME=\"$(libzscanner_SONAME)\" \
 		-DROOTHINTS=\"$(ROOTHINTS)\" \
+		-DLIBEXT=\"$(LIBEXT)\" \
 		-DLUA_HAS_SETFUNCS="$(LUA_HAS_SETFUNCS)"
 kresd_DEPEND := $(libkres) $(contrib)
 kresd_LIBS := $(libkres_TARGET) $(contrib_TARGET) $(libknot_LIBS) \
@@ -48,7 +49,7 @@ endif
 $(eval $(call make_sbin,kresd,daemon,yes))
 
 # Targets
-date := $(shell date +%F -r NEWS)
+date := $(shell date -r NEWS +%F)
 daemon: $(kresd) $(kresd_DIST)
 daemon-install: kresd-install bindings-install
 ifneq ($(SED),)
diff --git a/daemon/engine.c b/daemon/engine.c
index 81c44f7..44f68c2 100644
--- a/daemon/engine.c
+++ b/daemon/engine.c
@@ -234,8 +234,8 @@ int engine_set_moduledir(struct engine *engine, const char *moduledir) {
 		 "if package._path == nil then package._path = package.path end\n"
 		 "package.path = '%1$s/?.lua;%1$s/?/init.lua;'..package._path\n"
 		 "if package._cpath == nil then package._cpath = package.cpath end\n"
-		 "package.cpath = '%1$s/?.so;'..package._cpath\n",
-		 new_moduledir);
+		 "package.cpath = '%1$s/?%2$s;'..package._cpath\n",
+		 new_moduledir, LIBEXT);
 
 	int ret = l_dobytecode(engine->L, l_paths, strlen(l_paths), "");
 	if (ret != 0) {
@@ -329,7 +329,10 @@ static int l_hint_root_file(lua_State *L)
 
 	const char *err = engine_hint_root_file(ctx, file);
 	if (err) {
-		lua_pushstring(L, err);
+		if (!file) {
+			file = ROOTHINTS;
+		}
+		lua_push_printf(L, "error when opening '%s': %s", file, err);
 		lua_error(L);
 	} else {
 		lua_pushboolean(L, true);
diff --git a/lib/utils.h b/lib/utils.h
index 23c895c..140cbde 100644
--- a/lib/utils.h
+++ b/lib/utils.h
@@ -26,6 +26,7 @@
 #include <libknot/packet/pkt.h>
 #include <libknot/rrset.h>
 #include <libknot/rrtype/rrsig.h>
+#include <lua.h>
 #include "lib/generic/map.h"
 #include "lib/generic/array.h"
 #include "lib/defines.h"
@@ -288,3 +289,14 @@ static inline uint16_t kr_rrset_type_maysig(const knot_rrset_t *rr)
 		type = knot_rrsig_type_covered(&rr->rrs, 0);
 	return type;
 }
+
+/** Printf onto the lua stack, avoiding additional copy (thin wrapper). */
+static inline const char *lua_push_printf(lua_State *L, const char *fmt, ...)
+{
+	va_list args;
+	va_start(args, fmt);
+	const char *ret = lua_pushvfstring(L, fmt, args);
+	va_end(args);
+	return ret;
+}
+
diff --git a/modules/hints/hints.c b/modules/hints/hints.c
index fdd86b2..2cc3481 100644
--- a/modules/hints/hints.c
+++ b/modules/hints/hints.c
@@ -569,7 +569,7 @@ static char* hint_root_file(void *env, struct kr_module *module, const char *arg
 {
 	struct engine *engine = env;
 	struct kr_context *ctx = &engine->resolver;
-	return (char *)/*const-cast*/engine_hint_root_file(ctx, args);
+	return strdup(engine_hint_root_file(ctx, args));
 }
 
 /*
diff --git a/platform.mk b/platform.mk
index ad3367c..03669a1 100644
--- a/platform.mk
+++ b/platform.mk
@@ -9,7 +9,6 @@ GO := go
 CAT := cat
 SED := sed
 LIBEXT := .so
-MODEXT := $(LIBEXT)
 AREXT  := .a
 LIBTYPE := shared
 MODTYPE := shared
@@ -130,7 +129,6 @@ make_bin = $(call make_target,$(1),$(2),$(BINEXT),$(BINFLAGS),$(BINDIR),$(3))
 make_sbin = $(call make_target,$(1),$(2),$(BINEXT),$(BINFLAGS),$(SBINDIR),$(3))
 make_lib = $(call make_target,$(1),$(2),$(LIBEXT),-$(LIBTYPE),$(LIBDIR),$(3),$(4))
 make_module = $(call make_target,$(1),$(2),$(LIBEXT),-$(LIBTYPE),$(MODULEDIR),$(3))
-make_shared = $(call make_target,$(1),$(2),$(MODEXT),-$(MODTYPE),$(LIBDIR),$(3))
 make_static = $(call make_target,$(1),$(2),$(AREXT),-$(ARTYPE),$(LIBDIR),$(3))
 
 # Find library (pkg-config)
