diff --git a/configure.ac b/configure.ac
index 2330024..858dd1d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -111,6 +111,7 @@ AC_SUBST(LT_CURRENT_MINUS_AGE)
 dnl Checks for programs.
 AC_PROG_CC
 AC_PROG_CPP
+AC_PROG_OBJC
 AC_USE_SYSTEM_EXTENSIONS
 
 AM_CONDITIONAL(HAVE_GCC, [test "$GCC" = "yes"])
diff --git a/gio/Makefile.am b/gio/Makefile.am
index 67b7fa9..ee3de62 100644
--- a/gio/Makefile.am
+++ b/gio/Makefile.am
@@ -605,15 +605,14 @@ if OS_COCOA
 # We cannot simply rename the ObjC files to .m because it plays
 # tricks with Automake. For further reference, see:
 # https://bugzilla.gnome.org/show_bug.cgi?id=672777
-libgio_objc_2_0_la_CFLAGS = $(libgio_2_0_la_CFLAGS) -xobjective-c
 libgio_objc_2_0_la_CPPFLAGS = $(libgio_2_0_la_CPPFLAGS)
 libgio_objc_2_0_la_LDFLAGS = $(libgio_2_0_la_LDFLAGS) -Wl,-framework,Foundation -Wl,-framework,AppKit
 libgio_objc_2_0_la_SOURCES = \
-	gnextstepsettingsbackend.c \
-	gosxcontenttype.c \
-	gosxappinfo.c \
-	gosxappinfo.h \
-	gcocoanotificationbackend.c
+	gnextstepsettingsbackend.m \
+	gosxcontenttype.m \
+	gosxappinfo.m \
+	gosxappinfo.m \
+	gcocoanotificationbackend.m
 
 noinst_LTLIBRARIES += libgio-objc-2.0.la
 libgio_2_0_la_LIBADD += libgio-objc-2.0.la
