diff --git a/configure.ac b/configure.ac
index 02479b206..99618ff23 100644
--- a/configure.ac
+++ b/configure.ac
@@ -111,6 +111,7 @@ AC_SUBST(LT_CURRENT_MINUS_AGE)
 dnl Checks for programs.
 AC_PROG_CC
 AC_PROG_CPP
+AC_PROG_OBJC
 AC_USE_SYSTEM_EXTENSIONS
 
 AM_CONDITIONAL(HAVE_GCC, [test "$GCC" = "yes"])
diff --git a/gio/Makefile.am b/gio/Makefile.am
index 07506c08d..e5d9c19bd 100644
--- a/gio/Makefile.am
+++ b/gio/Makefile.am
@@ -135,7 +135,7 @@ endif
 
 if OS_COCOA
 cocoa_settings_sources = \
-	gnextstepsettingsbackend.c
+	gnextstepsettingsbackend.m
 
 settings_sources += $(cocoa_settings_sources)
 endif
@@ -282,7 +282,7 @@ appinfo_sources += $(unix_appinfo_sources)
 
 if OS_COCOA
 if MAC_OS_X_9
-unix_sources += gcocoanotificationbackend.c
+unix_sources += gcocoanotificationbackend.m
 endif
 endif
 
@@ -585,8 +585,6 @@ libgio_2_0_la_LDFLAGS = $(GLIB_LINK_FLAGS) \
 	-export-dynamic $(no_undefined)
 
 if OS_COCOA
-# This is dumb.  The ObjC source file should be properly named .m
-libgio_2_0_la_CFLAGS += -xobjective-c
 libgio_2_0_la_LDFLAGS += -Wl,-framework,Foundation -Wl,-framework,AppKit
 endif
 
