diff --git a/gio/Makefile.am b/gio/Makefile.am
index e993e2f..6eab656 100644
--- a/gio/Makefile.am
+++ b/gio/Makefile.am
@@ -131,7 +131,7 @@ endif
 
 if OS_COCOA
 settings_sources += \
-	gnextstepsettingsbackend.c
+	gnextstepsettingsbackend.m
 endif
 
 application_headers = \
@@ -516,8 +516,6 @@ libgio_2_0_la_LDFLAGS = $(GLIB_LINK_FLAGS) \
 	-export-dynamic $(no_undefined)
 
 if OS_COCOA
-# This is dumb.  The ObjC source file should be properly named .m
-libgio_2_0_la_CFLAGS += -xobjective-c
 libgio_2_0_la_LDFLAGS += -Wl,-framework,Foundation -Wl,-framework,AppKit
 endif
 
diff --git a/gio/Makefile.in b/gio/Makefile.in
index 5c0791f..e4fd4bd 100644
--- a/gio/Makefile.in
+++ b/gio/Makefile.in
@@ -133,7 +133,7 @@ TESTS = $(am__EXEEXT_2)
 @OS_WIN32_TRUE@	gregistrysettingsbackend.c
 
 @OS_COCOA_TRUE@am__append_18 = \
-@OS_COCOA_TRUE@	gnextstepsettingsbackend.c
+@OS_COCOA_TRUE@	gnextstepsettingsbackend.m
 
 @HAVE_INOTIFY_TRUE@am__append_19 = inotify
 @HAVE_INOTIFY_TRUE@am__append_20 = inotify/libinotify.la
@@ -160,8 +160,6 @@ TESTS = $(am__EXEEXT_2)
 @OS_WIN32_TRUE@am__append_34 = gwin32appinfo.c gwin32appinfo.h
 @OS_WIN32_TRUE@am__append_35 = -lshlwapi -lws2_32 -ldnsapi -liphlpapi
 
-# This is dumb.  The ObjC source file should be properly named .m
-@OS_COCOA_TRUE@am__append_36 = -xobjective-c
 @OS_COCOA_TRUE@am__append_37 = -Wl,-framework,Foundation -Wl,-framework,AppKit
 bin_PROGRAMS = gio-querymodules$(EXEEXT) glib-compile-schemas$(EXEEXT) \
 	glib-compile-resources$(EXEEXT) gsettings$(EXEEXT) \
@@ -305,7 +303,7 @@ am__libgio_2_0_la_SOURCES_DIST = gappinfo.c gappinfoprivate.h \
 	gsettingsschema.h gsettingsschema-internal.h gsettingsschema.c \
 	gsettings-mapping.h gsettings-mapping.c gsettings.c \
 	gregistrysettingsbackend.h gregistrysettingsbackend.c \
-	gnextstepsettingsbackend.c gdbusutils.h gdbusutils.c \
+	gnextstepsettingsbackend.m gdbusutils.h gdbusutils.c \
 	gdbusaddress.h gdbusaddress.c gdbusauthobserver.h \
 	gdbusauthobserver.c gdbusauth.h gdbusauth.c \
 	gdbusauthmechanism.h gdbusauthmechanism.c \
@@ -3438,12 +3436,12 @@ libgio_2_0_la-gregistrysettingsbackend.lo: gregistrysettingsbackend.c
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libgio_2_0_la_CPPFLAGS) $(CPPFLAGS) $(libgio_2_0_la_CFLAGS) $(CFLAGS) -c -o libgio_2_0_la-gregistrysettingsbackend.lo `test -f 'gregistrysettingsbackend.c' || echo '$(srcdir)/'`gregistrysettingsbackend.c
 
-libgio_2_0_la-gnextstepsettingsbackend.lo: gnextstepsettingsbackend.c
-@am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libgio_2_0_la_CPPFLAGS) $(CPPFLAGS) $(libgio_2_0_la_CFLAGS) $(CFLAGS) -MT libgio_2_0_la-gnextstepsettingsbackend.lo -MD -MP -MF $(DEPDIR)/libgio_2_0_la-gnextstepsettingsbackend.Tpo -c -o libgio_2_0_la-gnextstepsettingsbackend.lo `test -f 'gnextstepsettingsbackend.c' || echo '$(srcdir)/'`gnextstepsettingsbackend.c
+libgio_2_0_la-gnextstepsettingsbackend.lo: gnextstepsettingsbackend.m
+@am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libgio_2_0_la_CPPFLAGS) $(CPPFLAGS) $(libgio_2_0_la_CFLAGS) $(CFLAGS) -MT libgio_2_0_la-gnextstepsettingsbackend.lo -MD -MP -MF $(DEPDIR)/libgio_2_0_la-gnextstepsettingsbackend.Tpo -c -o libgio_2_0_la-gnextstepsettingsbackend.lo `test -f 'gnextstepsettingsbackend.m' || echo '$(srcdir)/'`gnextstepsettingsbackend.m
 @am__fastdepCC_TRUE@	$(AM_V_at)$(am__mv) $(DEPDIR)/libgio_2_0_la-gnextstepsettingsbackend.Tpo $(DEPDIR)/libgio_2_0_la-gnextstepsettingsbackend.Plo
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	$(AM_V_CC)source='gnextstepsettingsbackend.c' object='libgio_2_0_la-gnextstepsettingsbackend.lo' libtool=yes @AMDEPBACKSLASH@
+@AMDEP_TRUE@@am__fastdepCC_FALSE@	$(AM_V_CC)source='gnextstepsettingsbackend.m' object='libgio_2_0_la-gnextstepsettingsbackend.lo' libtool=yes @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libgio_2_0_la_CPPFLAGS) $(CPPFLAGS) $(libgio_2_0_la_CFLAGS) $(CFLAGS) -c -o libgio_2_0_la-gnextstepsettingsbackend.lo `test -f 'gnextstepsettingsbackend.c' || echo '$(srcdir)/'`gnextstepsettingsbackend.c
+@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libgio_2_0_la_CPPFLAGS) $(CPPFLAGS) $(libgio_2_0_la_CFLAGS) $(CFLAGS) -c -o libgio_2_0_la-gnextstepsettingsbackend.lo `test -f 'gnextstepsettingsbackend.m' || echo '$(srcdir)/'`gnextstepsettingsbackend.m
 
 libgio_2_0_la-gdbusutils.lo: gdbusutils.c
 @am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libgio_2_0_la_CPPFLAGS) $(CPPFLAGS) $(libgio_2_0_la_CFLAGS) $(CFLAGS) -MT libgio_2_0_la-gdbusutils.lo -MD -MP -MF $(DEPDIR)/libgio_2_0_la-gdbusutils.Tpo -c -o libgio_2_0_la-gdbusutils.lo `test -f 'gdbusutils.c' || echo '$(srcdir)/'`gdbusutils.c
diff --git a/gio/Makefile.am b/gio/Makefile.am
index 4d2569d..f9be7ff 100644
--- a/gio/Makefile.am
+++ b/gio/Makefile.am
@@ -266,7 +266,7 @@ unix_sources = \
 	$(NULL)
 
 if OS_COCOA
-unix_sources += gcocoanotificationbackend.c
+unix_sources += gcocoanotificationbackend.m
 endif
 
 giounixincludedir=$(includedir)/gio-unix-2.0/gio
diff --git a/gio/Makefile.in b/gio/Makefile.in
index 3c59e20..f5caca2 100644
--- a/gio/Makefile.in
+++ b/gio/Makefile.in
@@ -181,7 +181,7 @@ TESTS = $(am__EXEEXT_2)
 @OS_UNIX_TRUE@am__append_29 = gdesktopappinfo.c
 @OS_UNIX_TRUE@am__append_30 = xdgmime/libxdgmime.la
 @OS_UNIX_TRUE@am__append_31 = xdgmime/libxdgmime.la
-@OS_COCOA_TRUE@@OS_UNIX_TRUE@am__append_32 = gcocoanotificationbackend.c
+@OS_COCOA_TRUE@@OS_UNIX_TRUE@am__append_32 = gcocoanotificationbackend.m
 @HAVE_NETLINK_TRUE@@OS_UNIX_TRUE@am__append_33 = \
 @HAVE_NETLINK_TRUE@@OS_UNIX_TRUE@	gnetworkmonitornetlink.c \
 @HAVE_NETLINK_TRUE@@OS_UNIX_TRUE@	gnetworkmonitornetlink.h \
@@ -315,7 +315,7 @@ am__libgio_2_0_la_SOURCES_DIST = gappinfo.c gappinfoprivate.h \
 	gunixvolume.c gunixvolume.h gunixvolumemonitor.c \
 	gunixvolumemonitor.h gunixinputstream.c gunixoutputstream.c \
 	gcontenttype.c gcontenttypeprivate.h gfdonotificationbackend.c \
-	ggtknotificationbackend.c gcocoanotificationbackend.c \
+	ggtknotificationbackend.c gcocoanotificationbackend.m \
 	gnetworkmonitornetlink.c gnetworkmonitornetlink.h \
 	gnetworkmonitornm.c gnetworkmonitornm.h gdbusdaemon.c \
 	gdbusdaemon.h gdbus-daemon-generated.c \
@@ -3322,12 +3322,12 @@ libgio_2_0_la-ggtknotificationbackend.lo: ggtknotificationbackend.c
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libgio_2_0_la_CPPFLAGS) $(CPPFLAGS) $(libgio_2_0_la_CFLAGS) $(CFLAGS) -c -o libgio_2_0_la-ggtknotificationbackend.lo `test -f 'ggtknotificationbackend.c' || echo '$(srcdir)/'`ggtknotificationbackend.c
 
-libgio_2_0_la-gcocoanotificationbackend.lo: gcocoanotificationbackend.c
-@am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libgio_2_0_la_CPPFLAGS) $(CPPFLAGS) $(libgio_2_0_la_CFLAGS) $(CFLAGS) -MT libgio_2_0_la-gcocoanotificationbackend.lo -MD -MP -MF $(DEPDIR)/libgio_2_0_la-gcocoanotificationbackend.Tpo -c -o libgio_2_0_la-gcocoanotificationbackend.lo `test -f 'gcocoanotificationbackend.c' || echo '$(srcdir)/'`gcocoanotificationbackend.c
+libgio_2_0_la-gcocoanotificationbackend.lo: gcocoanotificationbackend.m
+@am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libgio_2_0_la_CPPFLAGS) $(CPPFLAGS) $(libgio_2_0_la_CFLAGS) $(CFLAGS) -MT libgio_2_0_la-gcocoanotificationbackend.lo -MD -MP -MF $(DEPDIR)/libgio_2_0_la-gcocoanotificationbackend.Tpo -c -o libgio_2_0_la-gcocoanotificationbackend.lo `test -f 'gcocoanotificationbackend.m' || echo '$(srcdir)/'`gcocoanotificationbackend.m
 @am__fastdepCC_TRUE@	$(AM_V_at)$(am__mv) $(DEPDIR)/libgio_2_0_la-gcocoanotificationbackend.Tpo $(DEPDIR)/libgio_2_0_la-gcocoanotificationbackend.Plo
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	$(AM_V_CC)source='gcocoanotificationbackend.c' object='libgio_2_0_la-gcocoanotificationbackend.lo' libtool=yes @AMDEPBACKSLASH@
+@AMDEP_TRUE@@am__fastdepCC_FALSE@	$(AM_V_CC)source='gcocoanotificationbackend.m' object='libgio_2_0_la-gcocoanotificationbackend.lo' libtool=yes @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libgio_2_0_la_CPPFLAGS) $(CPPFLAGS) $(libgio_2_0_la_CFLAGS) $(CFLAGS) -c -o libgio_2_0_la-gcocoanotificationbackend.lo `test -f 'gcocoanotificationbackend.c' || echo '$(srcdir)/'`gcocoanotificationbackend.c
+@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libgio_2_0_la_CPPFLAGS) $(CPPFLAGS) $(libgio_2_0_la_CFLAGS) $(CFLAGS) -c -o libgio_2_0_la-gcocoanotificationbackend.lo `test -f 'gcocoanotificationbackend.m' || echo '$(srcdir)/'`gcocoanotificationbackend.m
 
 libgio_2_0_la-gnetworkmonitornetlink.lo: gnetworkmonitornetlink.c
 @am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libgio_2_0_la_CPPFLAGS) $(CPPFLAGS) $(libgio_2_0_la_CFLAGS) $(CFLAGS) -MT libgio_2_0_la-gnetworkmonitornetlink.lo -MD -MP -MF $(DEPDIR)/libgio_2_0_la-gnetworkmonitornetlink.Tpo -c -o libgio_2_0_la-gnetworkmonitornetlink.lo `test -f 'gnetworkmonitornetlink.c' || echo '$(srcdir)/'`gnetworkmonitornetlink.c

