diff --git a/gio/gdbusprivate.c b/gio/gdbusprivate.c
index 1e8e1d6..5ad928e 100644
--- a/gio/gdbusprivate.c
+++ b/gio/gdbusprivate.c
@@ -2107,7 +2107,7 @@ _g_dbus_get_machine_id (GError **error)
   /* TODO: use PACKAGE_LOCALSTATEDIR ? */
   ret = NULL;
   first_error = NULL;
-  if (!g_file_get_contents ("/var/lib/dbus/machine-id",
+  if (!g_file_get_contents ("@@HOMEBREW_PREFIX@@/var/lib/dbus/machine-id",
                             &ret,
                             NULL,
                             &first_error) &&
@@ -2117,7 +2117,7 @@ _g_dbus_get_machine_id (GError **error)
                             NULL))
     {
       g_propagate_prefixed_error (error, first_error,
-                                  _("Unable to load /var/lib/dbus/machine-id or /etc/machine-id: "));
+                                  _("Unable to load @@HOMEBREW_PREFIX@@/var/lib/dbus/machine-id or /etc/machine-id: "));
     }
   else
     {
diff --git a/gio/xdgmime/xdgmime.c b/gio/xdgmime/xdgmime.c
index 2446d38..1042df2 100644
--- a/gio/xdgmime/xdgmime.c
+++ b/gio/xdgmime/xdgmime.c
@@ -235,7 +235,7 @@ xdg_init_dirs (void)
   xdg_data_dirs = getenv ("XDG_DATA_DIRS");
 
   if (xdg_data_dirs == NULL)
-    xdg_data_dirs = "/usr/local/share/:/usr/share/";
+    xdg_data_dirs = "@@HOMEBREW_PREFIX@@/share/:/usr/share/";
 
   /* Work out how many dirs we’re dealing with. */
   if (xdg_data_home != NULL || home != NULL)
diff --git a/glib/gutils.c b/glib/gutils.c
index 2e2d457..c97259b 100644
--- a/glib/gutils.c
+++ b/glib/gutils.c
@@ -2080,7 +2080,7 @@ g_build_system_data_dirs (void)
    */
 #ifndef G_OS_WIN32
   if (!data_dirs || !data_dirs[0])
-    data_dirs = "/usr/local/share/:/usr/share/";
+    data_dirs = "@@HOMEBREW_PREFIX@@/share/:/usr/share/";
 
   data_dir_vector = g_strsplit (data_dirs, G_SEARCHPATH_SEPARATOR_S, 0);
 #else

