diff --git a/gio/Makefile.am b/gio/Makefile.am
index 16c6387..22c2d34 100644
--- a/gio/Makefile.am
+++ b/gio/Makefile.am
@@ -255,9 +255,7 @@ endif
 if OS_UNIX
 platform_libadd += xdgmime/libxdgmime.la
 platform_deps += xdgmime/libxdgmime.la
-if !OS_COCOA
 appinfo_headers += gdesktopappinfo.h
-endif


 unix_sources = \
@@ -396,11 +394,9 @@ portal_sources = \
 	$(NULL)

 if OS_UNIX
-if !OS_COCOA
 contenttype_sources += gcontenttype.c
 appinfo_sources += gdesktopappinfo.c
 endif
-endif

 gio_base_sources =		\
 	gappinfo.c 		\
@@ -609,10 +605,7 @@ libgio_objc_2_0_la_CFLAGS = $(libgio_2_0_la_CFLAGS) -xobjective-c
 libgio_objc_2_0_la_CPPFLAGS = $(libgio_2_0_la_CPPFLAGS)
 libgio_objc_2_0_la_LDFLAGS = $(libgio_2_0_la_LDFLAGS) -Wl,-framework,Foundation -Wl,-framework,AppKit
 libgio_objc_2_0_la_SOURCES = \
-	gnextstepsettingsbackend.c \
-	gosxcontenttype.c \
-	gosxappinfo.c \
-	gosxappinfo.h
+	gnextstepsettingsbackend.c
 if MAC_OS_X_9
 libgio_objc_2_0_la_SOURCES += \
 	gcocoanotificationbackend.c
@@ -893,7 +886,6 @@ gdbus_LDADD = libgio-2.0.la 				\
 	$(NULL)

 if OS_UNIX
-if !OS_COCOA
 # ------------------------------------------------------------------------
 # gapplication(1) tool
 bin_PROGRAMS += gapplication
@@ -904,7 +896,6 @@ gapplication_LDADD = libgio-2.0.la 			\
 	$(top_builddir)/glib/libglib-2.0.la		\
 	$(NULL)
 endif
-endif

 completiondir = $(datadir)/bash-completion/completions
 completion_DATA = \
diff --git a/gio/giomodule.c b/gio/giomodule.c
index 1ead84c..cb9f915 100644
--- a/gio/giomodule.c
+++ b/gio/giomodule.c
@@ -43,12 +43,9 @@
 #endif
 #include <glib/gstdio.h>
 
-#if defined(G_OS_UNIX) && !defined(HAVE_COCOA)
+#if defined(G_OS_UNIX)
 #include "gdesktopappinfo.h"
 #endif
-#ifdef HAVE_COCOA
-#include "gosxappinfo.h"
-#endif
 
 #ifdef HAVE_COCOA
 #include <AvailabilityMacros.h>
@@ -1011,7 +1008,7 @@ _g_io_modules_ensure_extension_points_registered (void)
     {
       registered_extensions = TRUE;
       
-#if defined(G_OS_UNIX) && !defined(HAVE_COCOA)
+#if defined(G_OS_UNIX)
 #if !GLIB_CHECK_VERSION (3, 0, 0)
       ep = g_io_extension_point_register (G_DESKTOP_APP_INFO_LOOKUP_EXTENSION_POINT_NAME);
       G_GNUC_BEGIN_IGNORE_DEPRECATIONS
@@ -1152,7 +1149,6 @@ _g_io_modules_ensure_loaded (void)
 #endif
 #ifdef HAVE_COCOA
       g_type_ensure (g_nextstep_settings_backend_get_type ());
-      g_type_ensure (g_osx_app_info_get_type ());
 #endif
 #ifdef G_OS_UNIX
       g_type_ensure (_g_unix_volume_monitor_get_type ());
diff --git a/gio/glocalfileinfo.c b/gio/glocalfileinfo.c
index 50d870f..8fe3ae3 100644
--- a/gio/glocalfileinfo.c
+++ b/gio/glocalfileinfo.c
@@ -1290,7 +1290,7 @@ get_content_type (const char          *basename,
 
       content_type = g_content_type_guess (basename, NULL, 0, &result_uncertain);
       
-#if !defined(G_OS_WIN32) && !defined(HAVE_COCOA)
+#if !defined(G_OS_WIN32)
       if (!fast && result_uncertain && path != NULL)
 	{
 	  guchar sniff_buffer[4096];
diff --git a/gio/tests/Makefile.am b/gio/tests/Makefile.am
index 14cd928..722ff8f 100644
--- a/gio/tests/Makefile.am
+++ b/gio/tests/Makefile.am
@@ -347,13 +347,11 @@ uninstalled_test_extra_programs += \
 	gdbus-example-unix-fd-client		\
 	$(NULL)
 
-if !OS_COCOA
 test_extra_programs += apps
 test_programs += mimeapps
 clean-local: clean-mimeapps
 clean-mimeapps:
 	rm -rf xdgdatadir xdgdatahome xdgconfighome
-endif
 
 uninstalled_test_programs += gsettings gschema-compile
 gsettings_DEPENDENCIES = test.mo
