diff --git a/gio/Makefile.am b/gio/Makefile.am
index 8a0c2c2..8e24e96 100644
--- a/gio/Makefile.am
+++ b/gio/Makefile.am
@@ -3,10 +3,8 @@ include $(top_srcdir)/glib.mk
 SUBDIRS = gdbus-2.0/codegen
 
 if OS_UNIX
-if !OS_COCOA
 SUBDIRS += xdgmime
 endif
-endif
 
 if OS_WIN32_AND_DLL_COMPILATION
 if MS_LIB_AVAILABLE
@@ -251,12 +249,10 @@ SUBDIRS += fam
 endif
 
 if OS_UNIX
-if !OS_COCOA
 platform_libadd += xdgmime/libxdgmime.la
 platform_deps += xdgmime/libxdgmime.la
 
 appinfo_headers += gdesktopappinfo.h
-endif
 
 
 unix_sources = \
@@ -393,17 +389,10 @@ portal_sources = \
 	$(xdp_dbus_built_sources)	\
 	$(NULL)
 
-if OS_COCOA
-contenttype_sources += gosxcontenttype.c
-appinfo_sources += gosxappinfo.c gosxappinfo.h
-endif
-
 if OS_UNIX
-if !OS_COCOA
 contenttype_sources += gcontenttype.c
 appinfo_sources += gdesktopappinfo.c
 endif
-endif
 
 gio_base_sources =		\
 	gappinfo.c 		\
@@ -878,7 +867,6 @@ gdbus_LDADD = libgio-2.0.la 				\
 	$(NULL)
 
 if OS_UNIX
-if !OS_COCOA
 # ------------------------------------------------------------------------
 # gapplication(1) tool
 bin_PROGRAMS += gapplication
@@ -889,7 +877,6 @@ gapplication_LDADD = libgio-2.0.la 			\
 	$(top_builddir)/glib/libglib-2.0.la		\
 	$(NULL)
 endif
-endif
 
 completiondir = $(datadir)/bash-completion/completions
 completion_DATA = \
diff --git a/gio/giomodule.c b/gio/giomodule.c
index 7b9662d..9c584c6 100644
--- a/gio/giomodule.c
+++ b/gio/giomodule.c
@@ -43,12 +43,9 @@
 #endif
 #include <glib/gstdio.h>
 
-#if defined(G_OS_UNIX) && !defined(HAVE_COCOA)
+#if defined(G_OS_UNIX)
 #include "gdesktopappinfo.h"
 #endif
-#ifdef HAVE_COCOA
-#include "gosxappinfo.h"
-#endif
 
 #ifdef HAVE_COCOA
 #include <AvailabilityMacros.h>
@@ -978,7 +975,7 @@ _g_io_modules_ensure_extension_points_registered (void)
     {
       registered_extensions = TRUE;
       
-#if defined(G_OS_UNIX) && !defined(HAVE_COCOA)
+#ifdef G_OS_UNIX
 #if !GLIB_CHECK_VERSION (3, 0, 0)
       ep = g_io_extension_point_register (G_DESKTOP_APP_INFO_LOOKUP_EXTENSION_POINT_NAME);
       G_GNUC_BEGIN_IGNORE_DEPRECATIONS
@@ -1119,7 +1116,6 @@ _g_io_modules_ensure_loaded (void)
 #endif
 #ifdef HAVE_COCOA
       g_type_ensure (g_nextstep_settings_backend_get_type ());
-      g_type_ensure (g_osx_app_info_get_type ());
 #endif
 #ifdef G_OS_UNIX
       g_type_ensure (_g_unix_volume_monitor_get_type ());
diff --git a/gio/glocalfileinfo.c b/gio/glocalfileinfo.c
index 113a20b..fad29cf 100644
--- a/gio/glocalfileinfo.c
+++ b/gio/glocalfileinfo.c
@@ -1259,7 +1259,7 @@ get_content_type (const char          *basename,
 
       content_type = g_content_type_guess (basename, NULL, 0, &result_uncertain);
       
-#if !defined(G_OS_WIN32) && !defined(HAVE_COCOA)
+#if !defined(G_OS_WIN32)
       if (!fast && result_uncertain && path != NULL)
 	{
 	  guchar sniff_buffer[4096];
diff --git a/gio/tests/Makefile.am b/gio/tests/Makefile.am
index 59f2468..7b74bcc 100644
--- a/gio/tests/Makefile.am
+++ b/gio/tests/Makefile.am
@@ -324,13 +324,11 @@ uninstalled_test_extra_programs += \
 	gdbus-example-unix-fd-client		\
 	$(NULL)
 
-if !OS_COCOA
 test_extra_programs += apps
 test_programs += mimeapps
 clean-local: clean-mimeapps
 clean-mimeapps:
 	rm -rf xdgdatadir xdgdatahome xdgconfighome
-endif
 
 uninstalled_test_programs += gsettings gschema-compile
 gsettings_DEPENDENCIES = test.mo
