diff -pur a/corelib/thread.cpp b/corelib/thread.cpp
--- a/corelib/thread.cpp	2015-12-13 12:28:25.000000000 +0100
+++ b/corelib/thread.cpp	2017-10-05 12:47:00.000000000 +0200
@@ -26,6 +26,7 @@
 #include <string.h>
 #include <stdarg.h>
 #include <limits.h>
+#include <sys/sysctl.h>
 
 #if _POSIX_PRIORITY_SCHEDULING > 0
 #include <sched.h>
@@ -1311,10 +1312,10 @@ size_t Thread::cache(void)
     free(buffer);
     return line_size;
 #elif defined(__APPLE__)
-    size_t line_size = 0;
-    size_t sizeof_line_size = sizeof(line_size);
-    sysctlbyname("hw.cachelinesize", &line_size, &sizeof_line_size, 0, 0);
-    return line_size;
+    size_t lsize = 0;
+    size_t sizeof_line_size = sizeof(lsize);
+    sysctlbyname("hw.cachelinesize", &lsize, &sizeof_line_size, 0, 0);
+    return lsize;
 #elif defined(_SC_LEVEL1_DCACHE_LINESIZE)
     line_size = sysconf(_SC_LEVEL1_DCACHE_LINESIZE);
     if(!line_size)
