diff --git a/src/layout_util.c b/src/layout_util.c
index 0f164b5..66d4a42 100644
--- a/src/layout_util.c
+++ b/src/layout_util.c
@@ -477,6 +477,7 @@ static void layout_menu_write_rotate(GtkToggleAction *action, gpointer data, gbo
 	gint run_result;
 	GenericDialog *gd;
 	GString *message;
+	int cmdstatus;
 
 	if (!layout_valid(&lw)) return;
 
@@ -512,7 +513,8 @@ static void layout_menu_write_rotate(GtkToggleAction *action, gpointer data, gbo
 		command = g_strconcat(GQ_BIN_DIR, "/geeqie-rotate -r ", rotation,
 													keep_date ? " -t " : " ", fd_n->path, NULL);
 
-		run_result = WEXITSTATUS(runcmd(command));
+		cmdstatus = runcmd(command);
+		run_result = WEXITSTATUS(cmdstatus);
 		if (!run_result)
 			{
 			fd_n->user_orientation = 0;
