diff --git a/plugins/fmhttp/Makefile.in b/plugins/fmhttp/Makefile.in
index 394c94b..7b3d6f1 100644
--- a/plugins/fmhttp/Makefile.in
+++ b/plugins/fmhttp/Makefile.in
@@ -411,7 +411,7 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 pkglib_LTLIBRARIES = fmhttp.la
 fmhttp_la_SOURCES = fmhttp.c
-fmhttp_la_CPPFLAGS = -I$(top_srcdir) $(PTHREADS_CFLAGS) $(RSRT_CFLAGS) $(CURL_CFLAGS)
+fmhttp_la_CPPFLAGS = -I$(top_srcdir) $(PTHREADS_CFLAGS) $(RSRT_CFLAGS) $(CURL_CFLAGS) $(LIBLOGGING_STDLOG_CFLAGS)
 fmhttp_la_LDFLAGS = -module -avoid-version
 fmhttp_la_LIBADD = $(CURL_LIBS)
 all: all-am
diff --git a/plugins/imfile/Makefile.in b/plugins/imfile/Makefile.in
index d48e2ce..c5b6092 100644
--- a/plugins/imfile/Makefile.in
+++ b/plugins/imfile/Makefile.in
@@ -410,7 +410,7 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 pkglib_LTLIBRARIES = imfile.la
 imfile_la_SOURCES = imfile.c
-imfile_la_CPPFLAGS = -I$(top_srcdir) $(PTHREADS_CFLAGS) $(RSRT_CFLAGS)
+imfile_la_CPPFLAGS = -I$(top_srcdir) $(PTHREADS_CFLAGS) $(RSRT_CFLAGS) $(LIBLOGGING_STDLOG_CFLAGS)
 imfile_la_LDFLAGS = -module -avoid-version
 imfile_la_LIBADD = 
 all: all-am
