--- addons/AppleSensors/CMakeLists.txt.orig  2015-10-22 21:36:35.000000000 +0100
+++ addons/AppleSensors/CMakeLists.txt   2016-06-19 11:37:06.000000000 +0100
@@ -24,13 +24,14 @@
 )
 
 find_library(IOKIT IOKit)
+find_library(FOUNDATION Foundation)
 if (IOKIT)
   #add_executable(IoAppleSensors ${program_SOURCES})
 
   # Now build the shared library
   add_library(IoAppleSensors SHARED ${SRCS})
   add_dependencies(IoAppleSensors iovmall)
-  target_link_libraries(IoAppleSensors iovmall ${IOKIT})
+  target_link_libraries(IoAppleSensors iovmall ${IOKIT} ${FOUNDATION})
 
   # Install the addon to our global addons hierarchy.
   install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} DESTINATION lib/io/addons)
