--- a/scripts/airodump-ng-oui-update
+++ b/scripts/airodump-ng-oui-update
@@ -4,41 +4,9 @@
 WGET=`which wget 2>/dev/null`
 OUI_DOWNLOAD_URL="http://standards-oui.ieee.org/oui.txt"
 
-OUI_PATH0="/etc/aircrack-ng"
-OUI_PATH1="/usr/local/etc/aircrack-ng"
-OUI_PATH2="/usr/share/aircrack-ng"
-if [ -d "$OUI_PATH0" ]; then
-	OUI_PATH="$OUI_PATH0"
-elif [ -d "$OUI_PATH1" ]; then
-	OUI_PATH="$OUI_PATH1"
-elif [ -d "$OUI_PATH2" ]; then
-	OUI_PATH="$OUI_PATH2"
-else
-	# default
-	OUI_PATH="$OUI_PATH0"
-fi
-
+OUI_PATH="/usr/local/etc/aircrack-ng"
 AIRODUMP_NG_OUI="${OUI_PATH}/airodump-ng-oui.txt"
 OUI_IEEE="${OUI_PATH}/oui.txt"
-USERID=""
-
-
-# Make sure the user is root
-if [ x"`which id 2> /dev/null`" != "x" ]
-then
-	USERID="`id -u 2> /dev/null`"
-fi
-
-if [ x$USERID = "x" -a x$UID != "x" ]
-then
-	USERID=$UID
-fi
-
-if [ x$USERID != "x" -a x$USERID != "x0" ]
-then
-	echo Run it as root ; exit ;
-fi
-
 
 if [ ! -d "${OUI_PATH}" ]; then
 	mkdir -p ${OUI_PATH}
