diff -pur a/config.cc b/config.cc
--- a/config.cc   2018-06-01 22:18:19.000000000 -0500
+++ b/config.cc   2018-06-01 22:18:02.000000000 -0500
@@ -3258,7 +3258,7 @@

     fprintf(fp, ", biosdetect=%s", SIM->get_param_enum("biosdetect", base)->get_selected());

-    if (SIM->get_param_string("model", base)->getptr()>0) {
+    if (SIM->get_param_string("model", base)->getptr() != 0) {
         fprintf(fp, ", model=\"%s\"", SIM->get_param_string("model", base)->getptr());
     }
