diff -pur a/iodev/hdimage/cdrom_osx.cc b/iodev/hdimage/cdrom_osx.cc
--- a/iodev/hdimage/cdrom_osx.cc	2014-08-23 16:46:31.000000000 +0200
+++ b/iodev/hdimage/cdrom_osx.cc	2017-09-01 23:36:52.000000000 +0200
@@ -191,7 +191,7 @@ static struct _CDTOC * ReadTOC(const cha
   mach_port_t port = 0;
   char *devname;
 
-  if ((devname = strrchr(devpath, '/')) != NULL) {
+  if ((devname = (char *) strrchr(devpath, '/')) != NULL) {
     ++devname;
   }
   else {
