--- trafshow.c.orig	2017-08-14 01:05:28.000000000 -0700
+++ trafshow.c	2017-08-14 01:05:22.000000000 -0700
@@ -808,8 +808,6 @@
 static void
 vers()
 {
-	extern char pcap_version[];
-
 	int hc = 0;
 #ifdef  HAVE_HAS_COLORS
 	initscr();
@@ -820,7 +818,7 @@
 	fprintf(stderr, "\n%s Version %s\ncompiled for %s with\n %s\n",
 		progname, version, target, compiled);
 
-	fprintf(stderr, "\tlibpcap version %s\n", pcap_version);
+	fprintf(stderr, "\tlibpcap version %s\n", pcap_lib_version());
 
 #ifdef	HAVE_SLCURSES
 	fprintf(stderr, "\tslcurses version %d\n", SLang_Version);
