diff --git a/src/dssp.cpp b/src/dssp.cpp
index 29c1f82..55e364c 100755
--- a/src/dssp.cpp
+++ b/src/dssp.cpp
@@ -73,7 +73,7 @@ std::string ResidueToDSSPLine(const MResidue& residue)
 
   double alpha;
   char chirality;
-  std::tr1::tie(alpha,chirality) = residue.Alpha();
+  std::tie(alpha,chirality) = residue.Alpha();
 
   uint32 bp[2] = {};
   char bridgelabel[2] = { ' ', ' ' };
diff --git a/src/primitives-3d.cpp b/src/primitives-3d.cpp
index 3cfaa77..aa0014a 100755
--- a/src/primitives-3d.cpp
+++ b/src/primitives-3d.cpp
@@ -152,7 +152,7 @@ double CosinusAngle(const MPoint& p1, const MPoint& p2, const MPoint& p3, const
 
 // --------------------------------------------------------------------
 
-std::tr1::tuple<double,MPoint> QuaternionToAngleAxis(MQuaternion q)
+std::tuple<double,MPoint> QuaternionToAngleAxis(MQuaternion q)
 {
   if (q.R_component_1() > 1)
     q = Normalize(q);
@@ -168,7 +168,7 @@ std::tr1::tuple<double,MPoint> QuaternionToAngleAxis(MQuaternion q)
 
   MPoint axis(q.R_component_2() / s, q.R_component_3() / s, q.R_component_4() / s);
 
-  return std::tr1::make_tuple(angle, axis);
+  return std::make_tuple(angle, axis);
 }
 
 MPoint CenterPoints(std::vector<MPoint>& points)
diff --git a/src/primitives-3d.h b/src/primitives-3d.h
index a169a17..0171cc5 100755
--- a/src/primitives-3d.h
+++ b/src/primitives-3d.h
@@ -12,7 +12,7 @@
 #pragma once
 
 #include <boost/math/quaternion.hpp>
-#include <boost/tr1/tuple.hpp>
+#include <tuple>
 
 #include <vector>
 
@@ -71,7 +71,7 @@ double CosinusAngle(const MPoint& p1, const MPoint& p2, const MPoint& p3,
 
 MQuaternion Normalize(MQuaternion q);
 
-std::tr1::tuple<double,MPoint> QuaternionToAngleAxis(MQuaternion q);
+std::tuple<double,MPoint> QuaternionToAngleAxis(MQuaternion q);
 MPoint Centroid(std::vector<MPoint>& points);
 MPoint CenterPoints(std::vector<MPoint>& points);
 MQuaternion AlignPoints(const std::vector<MPoint>& a,
diff --git a/src/structure.cpp b/src/structure.cpp
index a5aa9bc..2135b19 100755
--- a/src/structure.cpp
+++ b/src/structure.cpp
@@ -554,7 +554,7 @@ double MResidue::Psi() const
   return result;
 }
 
-std::tr1::tuple<double,char> MResidue::Alpha() const
+std::tuple<double,char> MResidue::Alpha() const
 {
   double alhpa = 360;
   char chirality = ' ';
@@ -571,7 +571,7 @@ std::tr1::tuple<double,char> MResidue::Alpha() const
     else
       chirality = '+';
   }
-  return std::tr1::make_tuple(alhpa, chirality);
+  return std::make_tuple(alhpa, chirality);
 }
 
 double MResidue::Kappa() const
diff --git a/src/structure.h b/src/structure.h
index 3df6958..2cfbb58 100755
--- a/src/structure.h
+++ b/src/structure.h
@@ -171,7 +171,7 @@ class MResidue
 
   double        Phi() const;
   double        Psi() const;
-  std::tr1::tuple<double,char>
+  std::tuple<double,char>
             Alpha() const;
   double        Kappa() const;
   double        TCO() const;
