diff --git a/fig2dev/dev/genps.c b/fig2dev/dev/genps.c
index bb7d295..a7f3b5c 100644
--- a/fig2dev/dev/genps.c
+++ b/fig2dev/dev/genps.c
@@ -134,8 +134,10 @@ static bool iso_text_exist(F_compound *ob);
 static bool ellipse_exist(F_compound *ob);
 static bool approx_spline_exist(F_compound *ob);
 static void draw_gridline(float x1, float y1, float x2, float y2);
+#ifdef HAVE_X11_XPM_H
 static void convert_xpm_colors(unsigned char cmap[3][MAXCOLORMAPSIZE],
 		XpmColor *coltabl, int ncols);
+#endif
 static void genps_itp_spline(F_spline *s);
 static void genps_ctl_spline(F_spline *s);
 
diff --git a/fig2dev/dev/gensvg.c b/fig2dev/dev/gensvg.c
index 19645f2..5e2b13e 100644
--- a/fig2dev/dev/gensvg.c
+++ b/fig2dev/dev/gensvg.c
@@ -243,6 +243,7 @@
 #include <strings.h>
 #include <math.h>
 //#include <ctype.h>
+#include <time.h>
 #include "bool.h"
 #include "pi.h"
 
