diff --git a/configure.ac b/configure.ac
index f5af2e1c..7d2fb49a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5,7 +5,7 @@ AC_PREREQ(2.50)
 
 AC_CONFIG_HEADERS([irssi-config.h])
 AC_CONFIG_MACRO_DIR([m4])
-AM_INIT_AUTOMAKE([1.9 no-define foreign subdir-objects])
+AM_INIT_AUTOMAKE([1.9 no-define foreign subdir-objects nostdinc])
 
 AM_SILENT_RULES([yes])
 
diff --git a/src/core/Makefile.am b/src/core/Makefile.am
index 4cc2226c..051d2ae5 100644
--- a/src/core/Makefile.am
+++ b/src/core/Makefile.am
@@ -1,6 +1,7 @@
 noinst_LIBRARIES = libcore.a
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core \
 	$(GLIB_CFLAGS) \
diff --git a/src/fe-common/core/Makefile.am b/src/fe-common/core/Makefile.am
index cf4e8ee3..41cde839 100644
--- a/src/fe-common/core/Makefile.am
+++ b/src/fe-common/core/Makefile.am
@@ -1,6 +1,7 @@
 noinst_LIBRARIES = libfe_common_core.a
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src -I$(top_srcdir)/src/core/ \
 	$(GLIB_CFLAGS) \
 	-DHELPDIR=\""$(datadir)/irssi/help"\" \
diff --git a/src/fe-common/irc/Makefile.am b/src/fe-common/irc/Makefile.am
index a5e95885..a5350fa5 100644
--- a/src/fe-common/irc/Makefile.am
+++ b/src/fe-common/irc/Makefile.am
@@ -3,6 +3,7 @@ SUBDIRS = dcc notifylist
 noinst_LIBRARIES = libfe_common_irc.a
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core/ \
 	-I$(top_srcdir)/src/irc/core/ \
diff --git a/src/fe-common/irc/dcc/Makefile.am b/src/fe-common/irc/dcc/Makefile.am
index e0ad6296..f56f5ff9 100644
--- a/src/fe-common/irc/dcc/Makefile.am
+++ b/src/fe-common/irc/dcc/Makefile.am
@@ -1,6 +1,7 @@
 noinst_LIBRARIES = libfe_irc_dcc.a
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core/ \
 	-I$(top_srcdir)/src/irc/core/ \
diff --git a/src/fe-common/irc/notifylist/Makefile.am b/src/fe-common/irc/notifylist/Makefile.am
index b990dcd2..679524b9 100644
--- a/src/fe-common/irc/notifylist/Makefile.am
+++ b/src/fe-common/irc/notifylist/Makefile.am
@@ -1,6 +1,7 @@
 noinst_LIBRARIES = libfe_irc_notifylist.a
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core/ \
 	-I$(top_srcdir)/src/irc/core/ \
diff --git a/src/fe-fuzz/Makefile.am b/src/fe-fuzz/Makefile.am
index f52d9c47..a6f9c035 100644
--- a/src/fe-fuzz/Makefile.am
+++ b/src/fe-fuzz/Makefile.am
@@ -6,6 +6,7 @@ bin_PROGRAMS = irssi-fuzz server-fuzz
 CCLD=$(CXX) $(CXXFLAGS)
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core/ \
 	-I$(top_srcdir)/src/irc/core/ \
diff --git a/src/fe-fuzz/fe-common/core/Makefile.am b/src/fe-fuzz/fe-common/core/Makefile.am
index 9aa39a84..d90846fe 100644
--- a/src/fe-fuzz/fe-common/core/Makefile.am
+++ b/src/fe-fuzz/fe-common/core/Makefile.am
@@ -4,6 +4,7 @@ bin_PROGRAMS = theme-load-fuzz
 CCLD=$(CXX) $(CXXFLAGS)
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core/ \
 	-I$(top_srcdir)/src/irc/core/ \
diff --git a/src/fe-fuzz/irc/core/Makefile.am b/src/fe-fuzz/irc/core/Makefile.am
index cb8b7e31..aa4cbed6 100644
--- a/src/fe-fuzz/irc/core/Makefile.am
+++ b/src/fe-fuzz/irc/core/Makefile.am
@@ -4,6 +4,7 @@ bin_PROGRAMS = event-get-params-fuzz
 CCLD=$(CXX) $(CXXFLAGS)
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core/ \
 	-I$(top_srcdir)/src/irc/core/ \
diff --git a/src/fe-none/Makefile.am b/src/fe-none/Makefile.am
index 1233fc91..baa0b3cb 100644
--- a/src/fe-none/Makefile.am
+++ b/src/fe-none/Makefile.am
@@ -1,6 +1,7 @@
 bin_PROGRAMS = botti
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core/ \
 	-I$(top_srcdir)/src/irc/core/ \
diff --git a/src/fe-text/Makefile.am b/src/fe-text/Makefile.am
index 448a9389..fe0ea876 100644
--- a/src/fe-text/Makefile.am
+++ b/src/fe-text/Makefile.am
@@ -1,6 +1,7 @@
 bin_PROGRAMS = irssi
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core/ \
 	-I$(top_srcdir)/src/fe-common/core/ \
diff --git a/src/irc/core/Makefile.am b/src/irc/core/Makefile.am
index c668faea..e4e1f361 100644
--- a/src/irc/core/Makefile.am
+++ b/src/irc/core/Makefile.am
@@ -1,6 +1,7 @@
 noinst_LIBRARIES = libirc_core.a
 
 AM_CPPFLAGS = \
+    -I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core \
 	-DSYSCONFDIR=\""$(sysconfdir)"\" \
diff --git a/src/irc/dcc/Makefile.am b/src/irc/dcc/Makefile.am
index cf1e2ff5..147e360f 100644
--- a/src/irc/dcc/Makefile.am
+++ b/src/irc/dcc/Makefile.am
@@ -1,6 +1,7 @@
 noinst_LIBRARIES = libirc_dcc.a
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core/ \
 	-I$(top_srcdir)/src/irc/core/ \
diff --git a/src/irc/flood/Makefile.am b/src/irc/flood/Makefile.am
index 1c4d786a..d2f103c5 100644
--- a/src/irc/flood/Makefile.am
+++ b/src/irc/flood/Makefile.am
@@ -1,6 +1,7 @@
 noinst_LIBRARIES = libirc_flood.a
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core/ \
 	-I$(top_srcdir)/src/irc/core/ \
diff --git a/src/irc/notifylist/Makefile.am b/src/irc/notifylist/Makefile.am
index abd9d18d..283c2076 100644
--- a/src/irc/notifylist/Makefile.am
+++ b/src/irc/notifylist/Makefile.am
@@ -1,6 +1,7 @@
 noinst_LIBRARIES = libirc_notifylist.a
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core/ \
 	-I$(top_srcdir)/src/irc/core/ \
diff --git a/src/irc/proxy/Makefile.am b/src/irc/proxy/Makefile.am
index 4eb73bb8..72b7922d 100644
--- a/src/irc/proxy/Makefile.am
+++ b/src/irc/proxy/Makefile.am
@@ -2,6 +2,7 @@ moduledir = $(libdir)/irssi/modules
 module_LTLIBRARIES = libirc_proxy.la
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core/ \
 	-I$(top_srcdir)/src/irc/core/ \
diff --git a/src/lib-config/Makefile.am b/src/lib-config/Makefile.am
index 06c7787c..e9fc76d4 100644
--- a/src/lib-config/Makefile.am
+++ b/src/lib-config/Makefile.am
@@ -1,6 +1,7 @@
 noinst_LIBRARIES = libirssi_config.a
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	$(GLIB_CFLAGS)
 
diff --git a/src/otr/Makefile.am b/src/otr/Makefile.am
index 7cc02d6d..a72029e9 100644
--- a/src/otr/Makefile.am
+++ b/src/otr/Makefile.am
@@ -8,6 +8,7 @@ EXTRA_LTLIBRARIES = \
 	libotr_core_static.la
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core/ \
 	-I$(top_srcdir)/src/irc/core/ \
diff --git a/src/perl/Makefile.am b/src/perl/Makefile.am
index db52744e..61b77720 100644
--- a/src/perl/Makefile.am
+++ b/src/perl/Makefile.am
@@ -15,6 +15,8 @@ perl-core.lo: irssi-core.pl.h
 perl-signals.lo: perl-signals-list.h
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
+	-I$(builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core \
 	-I$(top_srcdir)/src/fe-common/core \
diff --git a/tests/fe-common/core/Makefile.am b/tests/fe-common/core/Makefile.am
index f048e95c..5b9f603c 100644
--- a/tests/fe-common/core/Makefile.am
+++ b/tests/fe-common/core/Makefile.am
@@ -3,6 +3,7 @@ include $(top_srcdir)/utils/glib-tap.mk
 PACKAGE_STRING=fe-common/core
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core \
 	$(GLIB_CFLAGS)
diff --git a/tests/fe-text/Makefile.am b/tests/fe-text/Makefile.am
index 38979e14..8d7543dd 100644
--- a/tests/fe-text/Makefile.am
+++ b/tests/fe-text/Makefile.am
@@ -3,6 +3,7 @@ include $(top_srcdir)/utils/glib-tap.mk
 PACKAGE_STRING=fe-text
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core \
 	-I$(top_srcdir)/src/fe-common/core \
diff --git a/tests/irc/core/Makefile.am b/tests/irc/core/Makefile.am
index f60da03f..87e80c9a 100644
--- a/tests/irc/core/Makefile.am
+++ b/tests/irc/core/Makefile.am
@@ -3,6 +3,7 @@ include $(top_srcdir)/utils/glib-tap.mk
 PACKAGE_STRING=irc/core
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core \
 	-DSYSCONFDIR=\""$(sysconfdir)"\" \
diff --git a/tests/irc/flood/Makefile.am b/tests/irc/flood/Makefile.am
index 58c0d98d..97f7c946 100644
--- a/tests/irc/flood/Makefile.am
+++ b/tests/irc/flood/Makefile.am
@@ -3,6 +3,7 @@ include $(top_srcdir)/utils/glib-tap.mk
 PACKAGE_STRING=irc/flood
 
 AM_CPPFLAGS = \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/core \
 	-DSYSCONFDIR=\""$(sysconfdir)"\" \
