diff --git a/Makefile.am b/Makefile.am
index ae9e383..9ac9082 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -7,8 +7,10 @@
 SUBDIRS = \
 	usbhid-dump
 
+if HAVE_GNULD
 AM_LDFLAGS = \
 	-Wl,--as-needed
+endif
 
 data_DATA =
 
diff --git a/configure.ac b/configure.ac
index 6b7fdb7..fca0254 100644
--- a/configure.ac
+++ b/configure.ac
@@ -19,7 +19,23 @@ PKG_INSTALLDIR
 
 PKG_CHECK_MODULES(LIBUSB, libusb-1.0 >= 1.0.14)
 
-PKG_CHECK_MODULES(UDEV, libudev >= 196)
+PKG_CHECK_MODULES(UDEV, libudev >= 196,
+		  [AC_DEFINE([HAVE_UDEV], [1], [Use UDEV])],
+		  [true])
+
+AC_MSG_CHECKING(for GNU ld)
+ac_prog=ld
+if test "$GCC" = yes; then
+       ac_prog=`$CC -print-prog-name=ld`
+fi
+case `"$ac_prog" -V 2>&1 < /dev/null` in
+      *GNU*)
+          GNULD=yes;;
+      *)
+          GNULD=no;;
+esac
+AC_MSG_RESULT($GNULD)
+AM_CONDITIONAL([HAVE_GNULD], [test "$GNULD" = yes])
 
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_FILES([
diff --git a/lsusb-t.c b/lsusb-t.c
index 81cf28a..f2d3e42 100644
--- a/lsusb-t.c
+++ b/lsusb-t.c
@@ -20,6 +20,8 @@
 #define MY_PATH_MAX 4096
 #define MY_PARAM_MAX 64
 
+#ifdef __linux__
+
 struct usbinterface {
 	struct list_head list;
 	struct usbinterface *next;
@@ -736,3 +738,11 @@ int lsusb_t(void)
 		perror(sys_bus_usb_devices);
 	return sbud == NULL;
 }
+
+#else
+int lsusb_t(void)
+{
+	fprintf(stderr, "lsusb -t is only supported on Linux\n");
+	return 1;
+}
+#endif /* __linux__ */
\ No newline at end of file
diff --git a/lsusb.c b/lsusb.c
index 5c64f42..24274db 100644
--- a/lsusb.c
+++ b/lsusb.c
@@ -3600,6 +3600,7 @@ static void dumpdev(libusb_device *dev)
 
 static int dump_one_device(libusb_context *ctx, const char *path)
 {
+#if __linux__	
 	libusb_device *dev;
 	struct libusb_device_descriptor desc;
 	char vendor[128], product[128];
@@ -3618,6 +3619,10 @@ static int dump_one_device(libusb_context *ctx, const char *path)
 					       product);
 	dumpdev(dev);
 	return 0;
+#else
+	fprintf(stderr, "Only supported on Linux\n");
+	return 1;
+#endif /* __linux__ */	
 }
 
 static void sort_device_list(libusb_device **list, ssize_t num_devs)
diff --git a/names.c b/names.c
index 7bc0ae9..98cb5c7 100644
--- a/names.c
+++ b/names.c
@@ -23,7 +23,9 @@
 #include <stdbool.h>
 
 #include <libusb.h>
-#include <libudev.h>
+#ifdef HAVE_UDEV
+# include <libudev.h>
+#endif
 
 #include "usb-spec.h"
 #include "names.h"
@@ -46,8 +48,10 @@ static unsigned int hashnum(unsigned int num)
 
 /* ---------------------------------------------------------------------- */
 
+#ifdef HAVE_UDEV
 static struct udev *udev = NULL;
 static struct udev_hwdb *hwdb = NULL;
+#endif
 static struct audioterminal *audioterminals_hash[HASHSZ] = { NULL, };
 static struct videoterminal *videoterminals_hash[HASHSZ] = { NULL, };
 static struct genericstrtable *hiddescriptors_hash[HASHSZ] = { NULL, };
@@ -116,9 +120,11 @@ static const char *hwdb_get(const char *modalias, const char *key)
 {
 	struct udev_list_entry *entry;
 
+#ifdef HAVE_UDEV
 	udev_list_entry_foreach(entry, udev_hwdb_get_properties_list_entry(hwdb, modalias, 0))
 		if (strcmp(udev_list_entry_get_name(entry), key) == 0)
 			return udev_list_entry_get_value(entry);
+#endif
 
 	return NULL;
 }
@@ -444,6 +450,7 @@ int names_init(void)
 {
 	int r;
 
+#ifdef HAVE_UDEV
 	udev = udev_new();
 	if (!udev)
 		r = -1;
@@ -452,6 +459,7 @@ int names_init(void)
 		if (!hwdb)
 			r = -1;
 	}
+#endif
 
 	r = hash_tables();
 
@@ -460,6 +468,8 @@ int names_init(void)
 
 void names_exit(void)
 {
+#ifdef HAVE_UDEV
 	hwdb = udev_hwdb_unref(hwdb);
 	udev = udev_unref(udev);
+#endif
 }
diff --git a/sysfs.c b/sysfs.c
index e073aa8..b6a9990 100644
--- a/sysfs.c
+++ b/sysfs.c
@@ -12,10 +12,11 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <stdio.h>
-#include <linux/limits.h>
-
 #include <libusb.h>
 
+#ifdef __linux__
+
+#include <linux/limits.h>
 #include "sysfs.h"
 
 /*
@@ -70,3 +71,14 @@ int read_sysfs_prop(char *buf, size_t size, char *sysfs_name, char *propname)
 	close(fd);
 	return n;
 }
+#else
+int get_sysfs_name(char *buf, size_t size, libusb_device *dev)
+{
+	return -1;
+}
+
+int read_sysfs_prop(char *buf, size_t size, char *sysfs_name, char *propname)
+{
+	return -1;
+}
+#endif /* __linux__ */
\ No newline at end of file
diff --git a/usbhid-dump/configure.ac b/usbhid-dump/configure.ac
index f07e47a..2180ebf 100644
--- a/usbhid-dump/configure.ac
+++ b/usbhid-dump/configure.ac
@@ -67,7 +67,7 @@ fi
 #
 # Checks for library functions.
 #
-AC_CHECK_FUNCS(libusb_set_option)
+AC_CHECK_FUNCS(libusb_set_option sigaction)
 
 #
 # Output
diff --git a/usbhid-dump/src/usbhid-dump.c b/usbhid-dump/src/usbhid-dump.c
index bd669e2..ab86085 100644
--- a/usbhid-dump/src/usbhid-dump.c
+++ b/usbhid-dump/src/usbhid-dump.c
@@ -912,7 +912,9 @@ main(int argc, char **argv)
     bool                dump_stream     = false;
     unsigned int        stream_timeout  = 60000;
 
+#ifdef HAVE_SIGACTION
     struct sigaction    sa;
+#endif
 
     /*
      * Extract program invocation name
@@ -1003,6 +1005,7 @@ main(int argc, char **argv)
     if (optind < argc)
         USAGE_ERROR("Positional arguments are not accepted");
 
+#ifdef HAVE_SIGACTION
     /*
      * Setup signal handlers
      */
@@ -1035,6 +1038,7 @@ main(int argc, char **argv)
     sigaction(SIGUSR1, &sa, NULL);
     sa.sa_handler = stream_resume_sighandler;
     sigaction(SIGUSR2, &sa, NULL);
+#endif /* HAVE_SIGACTION */
 
     /* Make stdout buffered - we will flush it explicitly */
     setbuf(stdout, NULL);
@@ -1043,6 +1047,7 @@ main(int argc, char **argv)
     result = run(dump_descriptor, dump_stream, stream_timeout,
                  bus_num, dev_addr, vid, pid, iface_num);
 
+#ifdef HAVE_SIGACTION
     /*
      * Restore signal handlers
      */
@@ -1060,7 +1065,7 @@ main(int argc, char **argv)
      */
     if (exit_signum != 0)
         raise(exit_signum);
-
+#endif /* HAVE_SIGACTION */
     return result;
 }
 
diff --git a/usbmisc.c b/usbmisc.c
index b12928f..1717ecd 100644
--- a/usbmisc.c
+++ b/usbmisc.c
@@ -25,6 +25,8 @@
 
 #include "usbmisc.h"
 
+#ifdef __linux__
+
 /* ---------------------------------------------------------------------- */
 
 static const char *devbususb = "/dev/bus/usb";
@@ -141,6 +143,8 @@ libusb_device *get_usb_device(libusb_context *ctx, const char *path)
 	return dev;
 }
 
+#endif /* __linux__ */
+
 static char *get_dev_string_ascii(libusb_device_handle *dev, size_t size,
                                   uint8_t id)
 {
diff --git a/usbreset.c b/usbreset.c
index 68f87cb..240f880 100644
--- a/usbreset.c
+++ b/usbreset.c
@@ -14,6 +14,9 @@
 #include <ctype.h>
 #include <limits.h>
 #include <dirent.h>
+
+#ifdef __linux__
+
 #include <sys/ioctl.h>
 
 #include <linux/usbdevice_fs.h>
@@ -191,3 +194,11 @@ int main(int argc, char **argv)
 	reset_device(dev);
 	return 0;
 }
+
+#else
+int main(int argc, char **argv)
+{
+	fprintf(stderr, "Only supported on Linux\n");
+	exit(1);
+}
+#endif /* __linux__ */
\ No newline at end of file
