diff --git a/plugin/gs/gvloadimage_gs.c b/plugin/gs/gvloadimage_gs.c
index 62dd3c9..6d9ed3d 100644
--- a/plugin/gs/gvloadimage_gs.c
+++ b/plugin/gs/gvloadimage_gs.c
@@ -32,6 +32,21 @@
 #include <ghostscript/ierrors.h>
 #include <cairo/cairo.h>
 
+
+/** Ensure compatibility with Ghostscipt versions newer than 9.16 **/
+
+#ifndef e_VMerror
+#define e_VMerror gs_error_VMerror
+#endif
+
+#ifndef e_unregistered
+#define e_unregistered gs_error_unregistered
+#endif
+
+#ifndef e_invalidid
+#define e_invalidid gs_error_invalidid
+#endif
+
 #ifdef WIN32
 #define NUL_FILE "nul"
 #else
