diff -pur gcc-11.3.0/gcc/config/darwin-driver.c gcc-11.3.0-patched/gcc/config/darwin-driver.c
--- gcc-11.3.0/gcc/config/darwin-driver.c	2022-04-21 09:58:52
+++ gcc-11.3.0-patched/gcc/config/darwin-driver.c	2022-12-12 12:19:08
@@ -64,7 +64,8 @@ validate_macosx_version_min (const char *version_str)
 
   major = strtoul (version_str, &end, 10);
 
-  if (major < 10 || major > 12 ) /* macOS 10, 11, and 12 are known. */
+  /* macOS 10, 11, and 12 are known. clang accepts up to 99.  */
+  if (major < 10 || major > 99)
     return NULL;
 
   /* Skip a separating period, if there's one.  */
diff -pur gcc-11.3.0/gcc/config.gcc gcc-11.3.0-patched/gcc/config.gcc
--- gcc-11.3.0/gcc/config.gcc	2022-04-21 09:58:52
+++ gcc-11.3.0-patched/gcc/config.gcc	2022-12-12 12:20:01
@@ -1872,7 +1872,7 @@ hppa[12]*-*-hpux11*)
 		dwarf2=no
 	fi
 	;;
-i[34567]86-*-darwin1[89]*)
+i[34567]86-*-darwin1[89]* | i[34567]86-*-darwin2*)
 	echo "Error: 32bit target is not supported after Darwin17" 1>&2
 	;;
 i[34567]86-*-darwin*)
@@ -1882,7 +1882,7 @@ i[34567]86-*-darwin*)
 	tmake_file="${tmake_file} ${cpu_type}/t-darwin32-biarch t-slibgcc"
 	tm_file="${tm_file} ${cpu_type}/darwin32-biarch.h"
 	;;
-x86_64-*-darwin1[89]* | x86_64-*-darwin2[01]*)
+x86_64-*-darwin1[89]* | x86_64-*-darwin2*)
 	# Only 64b from now
 	with_cpu=${with_cpu:-core2}
 	tmake_file="${tmake_file} t-slibgcc"
diff -pur gcc-11.3.0/gcc/configure gcc-11.3.0-patched/gcc/configure
--- gcc-11.3.0/gcc/configure	2022-04-21 09:59:35
+++ gcc-11.3.0-patched/gcc/configure	2022-12-12 12:19:08
@@ -27016,7 +27016,7 @@ $as_echo "$as_me: WARNING: LTO for $target requires bi
 	;;
     esac
     case $target_os in
-       darwin2[0-9]* | darwin19*)
+       darwin2* | darwin19*)
         { $as_echo "$as_me:${as_lineno-$LINENO}: checking assembler for llvm assembler x86-pad-for-align option" >&5
 $as_echo_n "checking assembler for llvm assembler x86-pad-for-align option... " >&6; }
 if ${gcc_cv_as_mllvm_x86_pad_for_align+:} false; then :
diff -pur gcc-11.3.0/gcc/configure.ac gcc-11.3.0-patched/gcc/configure.ac
--- gcc-11.3.0/gcc/configure.ac	2022-04-21 09:58:53
+++ gcc-11.3.0-patched/gcc/configure.ac	2022-12-12 12:19:08
@@ -4772,7 +4772,7 @@ foo:	nop
 	;;
     esac
     case $target_os in
-       darwin2[[0-9]]* | darwin19*)
+       darwin2* | darwin19*)
         gcc_GAS_CHECK_FEATURE([llvm assembler x86-pad-for-align option],
           gcc_cv_as_mllvm_x86_pad_for_align,,
           [-mllvm -x86-pad-for-align=false], [.text],,
