diff -ru a/mk/beforeauto.mk.in b/mk/beforeauto.mk.in
--- a/mk/beforeauto.mk.in	2018-05-09 01:57:00.000000000 +0900
+++ b/mk/beforeauto.mk.in	2021-11-10 09:41:14.000000000 +0900
@@ -1095,11 +1095,9 @@
 SharedLibrarySoNameTemplate = lib$$1$$2.$$3.$(SHAREDLIB_SUFFIX)
 SharedLibraryLibNameTemplate = lib$$1$$2.$(SHAREDLIB_SUFFIX)
 SharedLibraryPlatformLinkFlagsTemplate = -dynamiclib \
-                                         -install_name $(INSTALLLIBDIR)/$(SharedLibraryFullNameTemplate) \
-                                         -flat_namespace \
-                                         -undefined suppress
-PythonLibraryPlatformLinkFlagsTemplate = -bundle -flat_namespace \
-                                         -undefined suppress
+                                         -install_name $(INSTALLLIBDIR)/$(SharedLibraryFullNameTemplate)
+PythonLibraryPlatformLinkFlagsTemplate = -bundle \
+                                         -undefined dynamic_lookup
 PythonSHAREDLIB_SUFFIX = so
 
 
