diff --git a/vfe/unix/vfeplatform.h b/vfe/unix/vfeplatform.h
index d88a1481..17303ec8 100644
--- a/vfe/unix/vfeplatform.h
+++ b/vfe/unix/vfeplatform.h
@@ -36,6 +36,8 @@
 #ifndef __VFEPLATFORM_H__
 #define __VFEPLATFORM_H__
 
+#include <sys/types.h>
+
 #include <boost/shared_ptr.hpp>
 
 namespace vfePlatform
diff --git a/vfe/vfesession.h b/vfe/vfesession.h
index 7fd06a75..144bfe6b 100644
--- a/vfe/vfesession.h
+++ b/vfe/vfesession.h
@@ -36,6 +36,8 @@
 #ifndef __VFESESSION_H__
 #define __VFESESSION_H__
 
+#include <sys/types.h>
+
 #include <queue>
 
 #include "base/image/colourspace.h"
