diff --git a/unix/configure.ac b/unix/configure.ac
index 29c65c6e..c7bbf096 100644
--- a/unix/configure.ac
+++ b/unix/configure.ac
@@ -1002,8 +1002,8 @@ AC_DEFINE_UNQUOTED([BUILTIN_XWIN_DISPLAY], ["$pov_xwin_msg"], [X Window display.
 AC_DEFINE_UNQUOTED([BUILTIN_IMG_FORMATS], ["$pov_img_fmt_builtin $pov_img_fmt_enabled"], [Supported image formats.])
 AC_DEFINE_UNQUOTED([MISSING_IMG_FORMATS], ["$pov_img_fmt_disabled"], [Unsupported image formats.])
 
-povowner=`stat --format=%u ${HOME} || stat -f %u ${HOME}`
-povgroup=`stat --format=%g ${HOME} || stat -f %g ${HOME}`
+povowner=`stat -f %u ${HOME}`
+povgroup=`stat -f %g ${HOME}`
 AC_SUBST([povowner])
 AC_SUBST([povgroup])
 
