diff --git a/src/cc/error.h b/src/cc/error.h
index f6a3e6d..1b2dd35 100644
--- a/src/cc/error.h
+++ b/src/cc/error.h
@@ -37,10 +37,12 @@ namespace cc
 namespace error
 {
 
+static const char* strerror_result(int, const char* s) { return s; }
+
 static inline std::string strerror(const int ec)
 {
   char buf[256];
-  return strerror_r(ec, buf, sizeof(buf));
+  return strerror_result(strerror_r(ec, buf, sizeof(buf)), buf);
 }
 
 static inline std::string strerror() { return strerror(errno); }
diff --git a/src/cc/options.h b/src/cc/options.h
index 6a7e40c..fd7547b 100644
--- a/src/cc/options.h
+++ b/src/cc/options.h
@@ -747,7 +747,7 @@ private:
 #endif
               << "\n  built on " << BUILD_TIME
               << " for "         << CANONICAL_TARGET
-              << "\n    with "   << CXX", "CXXFLAGS
+              << "\n    with "   << CXX ", " CXXFLAGS
               << "\n  libquvi "  << quvi::version()
               << "\n  libquvi-scripts "
               << quvi_version(QUVI_VERSION_SCRIPTS)
