diff --git a/src/plugins/Visual/projectm/projectm.pro b/src/plugins/Visual/projectm/projectm.pro
index 0ea4180..2c4a5d0 100644
--- a/src/plugins/Visual/projectm/projectm.pro
+++ b/src/plugins/Visual/projectm/projectm.pro
@@ -4,31 +4,47 @@ QT += openglwidgets
 
 TARGET = $$PLUGINS_PREFIX/Visual/projectm
 
-HEADERS += projectmwidget.h \
-           visualprojectmfactory.h \
-           projectmplugin.h \
-           projectmwrapper.h
-SOURCES += projectmplugin.cpp \
-           visualprojectmfactory.cpp \
-           projectmwidget.cpp \
-           projectmwrapper.cpp
+HEADERS += visualprojectmfactory.h \
+           projectmplugin.h
+SOURCES += visualprojectmfactory.cpp \
+           projectmplugin.cpp
 
 RESOURCES = translations/translations.qrc
 
 #DEFINES += PROJECTM_31
+DEFINES += PROJECTM_4
+
+contains(DEFINES, PROJECTM_4) {
+    HEADERS += projectm4widget.h
+
+    SOURCES += projectm4widget.cpp
+} else {
+    HEADERS += projectmwidget.h \
+               projectmwrapper.h
+
+    SOURCES += projectmwidget.cpp \
+               projectmwrapper.cpp
+}
 
 unix {
-    PKGCONFIG += libprojectM
     target.path = $$PLUGIN_DIR/Visual
     INSTALLS += target
     LIBS += -L/usr/lib -I/usr/include
-    #projectM config path
-    PROJECTM_CONFIG_FILES = /usr/share/projectM/config.inp \
-                            /usr/local/share/projectM/config.inp
-    for(path, PROJECTM_CONFIG_FILES) {
-         exists($$path) {
-            message ("found projectm configuration: "$$path)
-           DEFINES += PROJECTM_CONFIG=\\\"$$path\\\"
+
+    contains(DEFINES, PROJECTM_4) {
+        INCLUDEPATH += /home/user/projectm/include
+        QMAKE_LIBDIR += /home/user/projectm/lib
+        LIBS += -lprojectM-4 -lprojectM-4-playlist
+    } else {
+        PKGCONFIG += libprojectM
+        #projectM config path
+        PROJECTM_CONFIG_FILES = /usr/share/projectM/config.inp \
+                                /usr/local/share/projectM/config.inp
+        for(path, PROJECTM_CONFIG_FILES) {
+            exists($$path) {
+                message("found projectm configuration: "$$path)
+                DEFINES += PROJECTM_CONFIG=\\\"$$path\\\"
+           }
         }
     }
 }
diff --git a/src/plugins/Visual/projectm/projectmplugin.cpp b/src/plugins/Visual/projectm/projectmplugin.cpp
index f8207cf..bba70be 100644
--- a/src/plugins/Visual/projectm/projectmplugin.cpp
+++ b/src/plugins/Visual/projectm/projectmplugin.cpp
@@ -25,13 +25,13 @@
 #include <QHBoxLayout>
 #include <QSplitter>
 #include <QListWidget>
-#include <math.h>
 #include <stdlib.h>
 #include <locale.h>
-#include <libprojectM/projectM.hpp>
-#include <qmmp/buffer.h>
-#include <qmmp/output.h>
+#ifdef PROJECTM_4
+#include "projectm4widget.h"
+#else
 #include "projectmwidget.h"
+#endif
 #include "projectmplugin.h"
 
 ProjectMPlugin::ProjectMPlugin (QWidget *parent)
@@ -45,7 +45,11 @@ ProjectMPlugin::ProjectMPlugin (QWidget *parent)
     QListWidget *listWidget = new QListWidget(m_splitter);
     listWidget->setAlternatingRowColors(true);
     m_splitter->addWidget(listWidget);
+#ifdef PROJECTM_4
+    m_projectMWidget = new ProjectM4Widget(listWidget, m_splitter);
+#else
     m_projectMWidget = new ProjectMWidget(listWidget, m_splitter);
+#endif
     m_splitter->addWidget(m_projectMWidget);
 
     m_splitter->setStretchFactor(1,1);
@@ -84,19 +88,9 @@ void ProjectMPlugin::stop()
 
 void ProjectMPlugin::onTimeout()
 {
-    projectM *instance = m_projectMWidget->projectMInstance();
-    if (!instance)
-        return;
-
     if(takeData(m_left, m_right))
     {
-        for(size_t i = 0; i < 512; i++)
-        {
-            m_buf[0][i] = m_left[i] * 32767.0;
-            m_buf[1][i] = m_right[i] * 32767.0;
-        }
-
-        m_projectMWidget->projectMInstance()->pcm()->addPCM16(m_buf);
+        m_projectMWidget->addPCM(m_left, m_right);
     }
 
     m_projectMWidget->update();
@@ -110,7 +104,7 @@ void ProjectMPlugin::setFullScreen(bool yes)
         setWindowState(windowState() & ~Qt::WindowFullScreen);
 }
 
-void ProjectMPlugin::closeEvent (QCloseEvent *event)
+void ProjectMPlugin::closeEvent(QCloseEvent *event)
 {
     //save geometry
     QSettings settings;
diff --git a/src/plugins/Visual/projectm/projectmplugin.h b/src/plugins/Visual/projectm/projectmplugin.h
index 0767b06..dd616d8 100644
--- a/src/plugins/Visual/projectm/projectmplugin.h
+++ b/src/plugins/Visual/projectm/projectmplugin.h
@@ -27,7 +27,11 @@
 class QTimer;
 class QSplitter;
 class Buffer;
+#ifdef PROJECTM_4
+class ProjectM4Widget;
+#else
 class ProjectMWidget;
+#endif
 
 class ProjectMPlugin : public Visual
 {
@@ -49,7 +53,11 @@ private:
     void showEvent(QShowEvent *) override;
     void hideEvent(QHideEvent *) override;
     QTimer *m_timer;
+#ifdef PROJECTM_4
+    ProjectM4Widget *m_projectMWidget;
+#else
     ProjectMWidget *m_projectMWidget;
+#endif
     QSplitter *m_splitter;
     short m_buf[2][512];
     float m_left[QMMP_VISUAL_NODE_SIZE];
diff --git a/src/plugins/Visual/projectm/projectmwidget.cpp b/src/plugins/Visual/projectm/projectmwidget.cpp
index b510037..182c4f9 100644
--- a/src/plugins/Visual/projectm/projectmwidget.cpp
+++ b/src/plugins/Visual/projectm/projectmwidget.cpp
@@ -51,9 +51,17 @@ ProjectMWidget::ProjectMWidget(QListWidget *listWidget, QWidget *parent)
 ProjectMWidget::~ProjectMWidget()
 {}
 
-projectM *ProjectMWidget::projectMInstance()
+void ProjectMWidget::addPCM(float *left, float *right)
 {
-    return m_projectM;
+    if(!m_projectM)
+        return;
+
+    for(size_t i = 0; i < qMin(512, QMMP_VISUAL_NODE_SIZE); i++)
+    {
+        m_buf[0][i] = left[i] * 32767.0;
+        m_buf[1][i] = right[i] * 32767.0;
+    }
+    m_projectM->pcm()->addPCM16(m_buf);
 }
 
 void ProjectMWidget::initializeGL()
diff --git a/src/plugins/Visual/projectm/projectmwidget.h b/src/plugins/Visual/projectm/projectmwidget.h
index 689d9d7..b3b4519 100644
--- a/src/plugins/Visual/projectm/projectmwidget.h
+++ b/src/plugins/Visual/projectm/projectmwidget.h
@@ -39,7 +39,7 @@ public:
 
     ~ProjectMWidget();
 
-    projectM *projectMInstance();
+    void addPCM(float *left, float *right);
 
 signals:
     void showMenuToggled(bool);
@@ -68,6 +68,7 @@ private:
     ProjectMWrapper *m_projectM;
     QMenu *m_menu;
     QListWidget *m_listWidget;
+    short m_buf[2][512];
 };
 
 #endif
diff --git a/src/plugins/Visual/projectm/visualprojectmfactory.cpp b/src/plugins/Visual/projectm/visualprojectmfactory.cpp
index e0f092c..1ccc01f 100644
--- a/src/plugins/Visual/projectm/visualprojectmfactory.cpp
+++ b/src/plugins/Visual/projectm/visualprojectmfactory.cpp
@@ -1,5 +1,5 @@
 /***************************************************************************
- *   Copyright (C) 2009-2015 by Ilya Kotov                                 *
+ *   Copyright (C) 2009-2023 by Ilya Kotov                                 *
  *   forkotov02@ya.ru                                                      *
  *                                                                         *
  *   This program is free software; you can redistribute it and/or modify  *
@@ -20,8 +20,8 @@
 
 #include <QMessageBox>
 #include <qmmp/qmmp.h>
-#include "visualprojectmfactory.h"
 #include "projectmplugin.h"
+#include "visualprojectmfactory.h"
 
 VisualProperties VisualProjectMFactory::properties() const
 {
diff --git a/src/plugins/Visual/projectm/visualprojectmfactory.h b/src/plugins/Visual/projectm/visualprojectmfactory.h
index 3770421..54c10b2 100644
--- a/src/plugins/Visual/projectm/visualprojectmfactory.h
+++ b/src/plugins/Visual/projectm/visualprojectmfactory.h
@@ -1,5 +1,5 @@
 /***************************************************************************
- *   Copyright (C) 2009-2014 by Ilya Kotov                                 *
+ *   Copyright (C) 2009-2023 by Ilya Kotov                                 *
  *   forkotov02@ya.ru                                                      *
  *                                                                         *
  *   This program is free software; you can redistribute it and/or modify  *
