diff --git a/copypackages.c b/copypackages.c
index 428ce13..096518b 100644
--- a/copypackages.c
+++ b/copypackages.c
@@ -445,7 +445,7 @@ static retvalue by_name(struct package_list *list, struct target *desttarget, st
 		struct package package;

 		for (prev = nameandversion ; prev < d ; prev++) {
-			if (strcmp(prev->name, d->name) == 0 && strcmp2(prev->version, d->version) == 0)
+			if (strcmp(prev->name, d->name) == 0 && strcmp(prev->version, d->version) == 0)
 				break;
 		}
 		if (prev < d) {
diff --git a/database.c b/database.c
index b01c8ed..ae82328 100644
--- a/database.c
+++ b/database.c
@@ -1079,7 +1079,7 @@ retvalue table_close(struct table *table) {
 	}

 	for (struct opened_tables *iter = opened_tables; iter != NULL; iter = iter->next) {
-		if(strcmp2(iter->name, table->name) == 0 && strcmp2(iter->subname, table->subname) == 0) {
+		if(strcmp(iter->name, table->name) == 0 && strcmp(iter->subname, table->subname) == 0) {
 			if (prev == NULL) {
 				opened_tables = iter->next;
 			} else {
@@ -1846,7 +1846,7 @@ static retvalue database_table_secondary(const char *filename, const char *subta
 		        filename, subtable, type, flags, secondary_filename, secondary_type);

 	for (struct opened_tables *iter = opened_tables; iter != NULL; iter = iter->next) {
-		if(strcmp2(iter->name, filename) == 0 && strcmp2(iter->subname, subtable) == 0) {
+		if(strcmp(iter->name, filename) == 0 && strcmp(iter->subname, subtable) == 0) {
             fprintf(stderr,
  "Internal Error: Trying to open table '%s' from file '%s' multiple times.\n"
  "This should normally not happen (to avoid triggering bugs in the underlying BerkeleyDB)\n",
