diff --git a/configure b/configure
index a752cd6..74e61d7 100755
--- a/configure
+++ b/configure
@@ -13,7 +13,7 @@ chomp($xmlldflags);

 my $prefix = "/usr/local";
 my $cflags = "-Wall -Wno-format-y2k -O2 -DLOCALEPATH=\"\\\"\$(LOCALEPATH)\\\"\" -DOS=\\\"$os\\\" $xmlcflags \$(EXTRA_CFLAGS) ";
-my $ldflags = "-lncurses -lcrypto $xmlldflags \$(EXTRA_LDFLAGS) ";
+my $ldflags = "\$(EXTRA_LDFLAGS) -lncurses -lcrypto $xmlldflags ";

 my $use_nls = 1;
 parse_cmdl_line();
