diff -pur PDFlib-Lite-7.0.5p3.old/libs/pdcore/pc_config.h PDFlib-Lite-7.0.5p3/libs/pdcore/pc_config.h
--- PDFlib-Lite-7.0.5p3.old/libs/pdcore/pc_config.h	2012-06-06 13:58:58.000000000 +0200
+++ PDFlib-Lite-7.0.5p3/libs/pdcore/pc_config.h	2021-12-30 11:28:15.000000000 +0100
@@ -142,8 +142,7 @@
 
 /* try to identify the Mac OS X command line compiler */
 
-#if defined(__APPLE__) && (defined(__ppc__) || \
-    defined(__i386__) || defined(__ppc64__) || defined(__x86_64__))
+#if defined(__APPLE__)
 
 #define MACOSX
 
