diff --git a/CONTRIBUTING.md b/CONTRIBUTING.md
new file mode 100644
index 0000000..3681694
--- /dev/null
+++ b/CONTRIBUTING.md
@@ -0,0 +1,27 @@
+# How to Report an Issue
+
+If you would like to report a problem you find when using `you-get`, please open a [Pull Request](https://github.com/soimort/you-get/pulls), which should include:
+
+1. A detailed description of the encountered problem;
+2. At least one commit, addressing the problem through some unit test(s).
+   * Examples of good commits: [#2675](https://github.com/soimort/you-get/pull/2675/files), [#2680](https://github.com/soimort/you-get/pull/2680/files), [#2685](https://github.com/soimort/you-get/pull/2685/files)
+
+PRs that fail to meet the above criteria may be closed summarily with no further action.
+
+A valid PR will remain open until its addressed problem is fixed.
+
+
+
+# 如何汇报问题
+
+为了防止对 GitHub Issues 的滥用，本项目不接受一般的 Issue。
+
+如您在使用 `you-get` 的过程中发现任何问题，请开启一个 [Pull Request](https://github.com/soimort/you-get/pulls)。该 PR 应当包含：
+
+1. 详细的问题描述；
+2. 至少一个 commit，其内容是**与问题相关的**单元测试。**不要通过随意修改无关文件的方式来提交 PR！**
+   * 有效的 commit 示例：[#2675](https://github.com/soimort/you-get/pull/2675/files), [#2680](https://github.com/soimort/you-get/pull/2680/files), [#2685](https://github.com/soimort/you-get/pull/2685/files)
+
+不符合以上条件的 PR 可能被直接关闭。
+
+有效的 PR 将会被一直保留，直至相应的问题得以修复。
diff --git a/contrib/completion/_you-get b/contrib/completion/_you-get
new file mode 100644
index 0000000..696aad8
--- /dev/null
+++ b/contrib/completion/_you-get
@@ -0,0 +1,29 @@
+#compdef you-get
+
+# Zsh completion definition for soimort/you-get.
+
+setopt localoptions noshwordsplit noksharrays
+local -a args
+
+args=(
+    '(- : *)'{-V,--version}'[print version and exit]'
+    '(- : *)'{-h,--help}'[print help and exit]'
+    '(-i --info)'{-i,--info}'[print extracted information]'
+    '(-u --url)'{-u,--url}'[print extracted information with URLs]'
+    '(--json)--json[print extracted URLs in JSON format]'
+    '(-n --no-merge)'{-n,--no-merge}'[do not merge video parts]'
+    '(--no-caption)--no-caption[do not download captions]'
+    '(-f --force)'{-f,--force}'[force overwrite existing files]'
+    '(-F --format)'{-F,--format}'[set video format to the specified stream id]:stream id'
+    '(-O --output-filename)'{-O,--output-filename}'[set output filename]:filename:_files'
+    '(-o --output-dir)'{-o,--output-dir}'[set output directory]:directory:_files -/'
+    '(-p --player)'{-p,--player}'[stream extracted URL to the specified player]:player and options'
+    '(-c --cookies)'{-c,--cookies}'[load cookies.txt or cookies.sqlite]:cookies file:_files'
+    '(-x --http-proxy)'{-x,--http-proxy}'[use the specified HTTP proxy for downloading]:host\:port:'
+    '(-y --extractor-proxy)'{-y,--extractor-proxy}'[use the specified HTTP proxy for extraction only]:host\:port'
+    '(--no-proxy)--no-proxy[do not use a proxy]'
+    '(-t --timeout)'{-t,--timeout}'[set socket timeout]:seconds'
+    '(-d --debug)'{-d,--debug}'[show traceback and other debug info]'
+    '*: :_guard "^-*" url'
+)
+_arguments -S -s $args
diff --git a/contrib/completion/you-get-completion.bash b/contrib/completion/you-get-completion.bash
new file mode 100755
index 0000000..9c6480e
--- /dev/null
+++ b/contrib/completion/you-get-completion.bash
@@ -0,0 +1,31 @@
+# Bash completion definition for you-get.
+
+_you-get () {
+    COMPREPLY=()
+    local IFS=$' \n'
+    local cur=$2 prev=$3
+    local -a opts_without_arg opts_with_arg
+    opts_without_arg=(
+        -V --version -h --help -i --info -u --url --json -n --no-merge
+        --no-caption -f --force --no-proxy -d --debug
+    )
+    opts_with_arg=(
+        -F --format -O --output-filename -o --output-dir -p --player
+        -c --cookies -x --http-proxy -y --extractor-proxy -t --timeout
+    )
+
+    # Do not complete non option names
+    [[ $cur == -* ]] || return 1
+
+    # Do not complete when the previous arg is an option expecting an argument
+    for opt in "${opts_with_arg[@]}"; do
+        [[ $opt == $prev ]] && return 1
+    done
+
+    # Complete option names
+    COMPREPLY=( $(compgen -W "${opts_without_arg[*]} ${opts_with_arg[*]}" \
+                          -- "$cur") )
+    return 0
+}
+
+complete -F _you-get you-get
diff --git a/contrib/completion/you-get.fish b/contrib/completion/you-get.fish
new file mode 100644
index 0000000..6917c42
--- /dev/null
+++ b/contrib/completion/you-get.fish
@@ -0,0 +1,23 @@
+# Fish completion definition for you-get.
+
+complete -c you-get -s V -l version -d 'print version and exit'
+complete -c you-get -s h -l help -d 'print help and exit'
+complete -c you-get -s i -l info -d 'print extracted information'
+complete -c you-get -s u -l url -d 'print extracted information'
+complete -c you-get -l json -d 'print extracted URLs in JSON format'
+complete -c you-get -s n -l no-merge -d 'do not merge video parts'
+complete -c you-get -l no-caption -d 'do not download captions'
+complete -c you-get -s f -l force -d 'force overwrite existing files'
+complete -c you-get -s F -l format -x -d 'set video format to the specified stream id'
+complete -c you-get -s O -l output-filename -d 'set output filename' \
+         -x -a '(__fish_complete_path (commandline -ct) "output filename")'
+complete -c you-get -s o -l output-dir  -d 'set output directory' \
+         -x -a '(__fish_complete_directories (commandline -ct) "output directory")'
+complete -c you-get -s p -l player -x -d 'stream extracted URL to the specified player'
+complete -c you-get -s c -l cookies -d 'load cookies.txt or cookies.sqlite' \
+         -x -a '(__fish_complete_path (commandline -ct) "cookies.txt or cookies.sqlite")'
+complete -c you-get -s x -l http-proxy -x -d 'use the specified HTTP proxy for downloading'
+complete -c you-get -s y -l extractor-proxy -x -d 'use the specified HTTP proxy for extraction only'
+complete -c you-get -l no-proxy -d 'do not use a proxy'
+complete -c you-get -s t -l timeout -x -d 'set socket timeout'
+complete -c you-get -s d -l debug -d 'show traceback and other debug info'
diff --git a/you-get.plugin.zsh b/you-get.plugin.zsh
new file mode 100644
index 0000000..6282eda
--- /dev/null
+++ b/you-get.plugin.zsh
@@ -0,0 +1,3 @@
+#!/usr/bin/env zsh
+alias you-get="noglob python3 $(dirname $0)/you-get"
+alias you-vlc="noglob python3 $(dirname $0)/you-get --player vlc"
