diff --git a/src/optipng/osys.c b/src/optipng/osys.c
index c35f5a5..66fd93d 100644
--- a/src/optipng/osys.c
+++ b/src/optipng/osys.c
@@ -508,7 +508,7 @@ osys_copy_attr(const char *src_path, const char *dest_path)
     if (chmod(dest_path, sbuf.st_mode) != 0)
         result = -1;
 
-#if defined AT_FDCWD && defined UTIME_NOW && defined UTIME_OMIT
+#if defined AT_FDCWD && defined UTIME_NOW && defined UTIME_OMIT && !defined __APPLE__
     {
         struct timespec times[2];
 
