diff --git a/source/Irrlicht/MacOSX/CIrrDeviceMacOSX.mm b/source/Irrlicht/MacOSX/CIrrDeviceMacOSX.mm
index a86c241..080a43d 100644
--- a/source/Irrlicht/MacOSX/CIrrDeviceMacOSX.mm
+++ b/source/Irrlicht/MacOSX/CIrrDeviceMacOSX.mm
@@ -731,7 +731,7 @@ bool CIrrDeviceMacOSX::createWindow()
 						{
 							[Window center];
 						}
-						[Window setDelegate:(id<NSWindowDelegate>)[NSApp delegate]];
+						[(NSFileManager *)Window setDelegate:(id<NSWindowDelegate>)[NSApp delegate]];
 
 						if(CreationParams.DriverType == video::EDT_OPENGL)
 							[OGLContext setView:[Window contentView]];
