diff -pur httperf-0.9.0/src/core.c httperf-0.9.0-fixed/src/core.c
--- httperf-0.9.0/src/core.c	2007-04-07 09:01:56.000000000 +0200
+++ httperf-0.9.0-fixed/src/core.c	2019-09-11 14:12:54.000000000 +0200
@@ -805,8 +805,8 @@ core_ssl_connect (Conn *s)
	fprintf (stderr, "core_ssl_connect: server refused all client cipher "
		 "suites!\n");
       else
-	fprintf (stderr, "core_ssl_connect: cipher=%s, valid=%d, id=%lu\n",
-		 ssl_cipher->name, ssl_cipher->valid, ssl_cipher->id);
+	fprintf (stderr, "core_ssl_connect: cipher=%s, id=%lu\n",
+		 SSL_CIPHER_get_name(ssl_cipher), SSL_CIPHER_get_id(ssl_cipher));
     }

   arg.l = 0;
diff -pur httperf-0.9.0/src/httperf.c httperf-0.9.0-fixed/src/httperf.c
--- httperf-0.9.0/src/httperf.c	2007-04-07 09:01:56.000000000 +0200
+++ httperf-0.9.0-fixed/src/httperf.c	2019-09-11 14:12:54.000000000 +0200
@@ -808,7 +808,7 @@ main (int argc, char **argv)
       SSLeay_add_ssl_algorithms ();

       /* for some strange reason, SSLv23_client_method () doesn't work here */
-      ssl_ctx = SSL_CTX_new (SSLv3_client_method ());
+      ssl_ctx = SSL_CTX_new (TLS_client_method ());
       if (!ssl_ctx)
	{
	  ERR_print_errors_fp (stderr);
