diff --git a/gtk/gtkmm/clipboard.cc b/gtk/gtkmm/clipboard.cc
index 52313e2..79ab14a 100644
--- a/gtk/gtkmm/clipboard.cc
+++ b/gtk/gtkmm/clipboard.cc
@@ -573,12 +573,6 @@ void Clipboard::store()
   gtk_clipboard_store(gobj());
 }
 
-GdkAtom Clipboard::get_selection() const
-{
-  return gtk_clipboard_get_selection(const_cast<GtkClipboard*>(gobj()));
-}
-
-
 Glib::SignalProxy< void,GdkEventOwnerChange* > Clipboard::signal_owner_change()
 {
   return Glib::SignalProxy< void,GdkEventOwnerChange* >(this, &Clipboard_signal_owner_change_info);
diff --git a/gtk/gtkmm/clipboard.h b/gtk/gtkmm/clipboard.h
index 798641c..a950445 100644
--- a/gtk/gtkmm/clipboard.h
+++ b/gtk/gtkmm/clipboard.h
@@ -557,14 +557,6 @@ public:
    * @newin{2,6}
    */
   void store();
-  
-  /** Gets the selection that this clipboard is for.
-   * 
-   * @newin{3,22}
-   * 
-   * @return The selection.
-   */
-  GdkAtom get_selection() const;
 
   
   /**
