diff --git a/doc/po4a.conf b/doc/po4a.conf
index 0798eac..1adc26b 100644
--- a/doc/po4a.conf
+++ b/doc/po4a.conf
@@ -4,11 +4,11 @@
 # Entities need to be present, even if not translated
 [po4a_alias:entity] text opt:"-k 0"
 [po4a_alias:manpage] docbook opt:"\
-	-o untranslated='<term><option> <refname> <refentrytitle> <date> <refentryinfo> <phrase> <variablelist> <term><command>' \
-	-o break='<term><option> <term><command>' \
-	-o nodefault='<option> <refname> <refentrytitle> <date> <refentryinfo> <phrase> <variablelist> <command>' \
-	-o inline='<para><option> <refentrytitle> <para><command> <programlisting><command> <refentrytitle><command> <refpurpose><command>' \
-	-o translated='<para><command> <programlisting><command> <refentrytitle><command> <refpurpose><command>'"
+	-o untranslated=\"<term><option> <refname> <refentrytitle> <date> <refentryinfo> <phrase> <variablelist> <term><command>\" \
+	-o break=\"<term><option> <term><command>\" \
+	-o nodefault=\"<option> <refname> <refentrytitle> <date> <refentryinfo> <phrase> <variablelist> <command>\" \
+	-o inline=\"<para><option> <refentrytitle> <para><command> <programlisting><command> <refentrytitle><command> <refpurpose><command>\" \
+	-o translated=\"<para><command> <programlisting><command> <refentrytitle><command> <refpurpose><command>\""
 
 # define source file and translated file (one file per line)
 [type: entity]  apt.ent $lang:$lang/apt.ent
