diff -pur julia-1.8.4/deps/csl.mk julia-1.8.4-patched/deps/csl.mk
--- julia-1.8.4/deps/csl.mk	2022-12-23 23:14:27
+++ julia-1.8.4-patched/deps/csl.mk	2022-12-31 12:28:13
@@ -75,10 +75,8 @@ else
 $(eval $(call copy_csl,$(call versioned_libname,libgcc_s_seh,1)))
 endif
 else
-ifeq ($(APPLE_ARCH),arm64)
-$(eval $(call copy_csl,$(call versioned_libname,libgcc_s,1.1)))
-else
-$(eval $(call copy_csl,$(call versioned_libname,libgcc_s,1)))
+ifeq ($(OS),Darwin)
+$(eval $(call symlink_system_library,CSL,libgcc_s,1.1))
 endif
 endif
 # winpthread is only Windows, pthread is only others
