diff --git a/Make.inc b/Make.inc
index 3fa153c..8e95675 100644
--- a/Make.inc
+++ b/Make.inc
@@ -1467,11 +1467,7 @@ LIBGCC_NAME := libgcc_s_seh-1.$(SHLIB_EXT)
 endif
 endif
 ifeq ($(OS),Darwin)
-ifeq ($(ARCH),aarch64)
 LIBGCC_NAME := libgcc_s.1.1.$(SHLIB_EXT)
-else
-LIBGCC_NAME := libgcc_s.1.$(SHLIB_EXT)
-endif
 endif
 ifneq ($(findstring $(OS),Linux FreeBSD),)
 LIBGCC_NAME := libgcc_s.$(SHLIB_EXT).1
diff --git a/base/Makefile b/base/Makefile
index 23a9c40..846b197 100644
--- a/base/Makefile
+++ b/base/Makefile
@@ -206,7 +206,7 @@ else ifneq ($(USE_SYSTEM_OPENLIBM),0)
 $(eval $(call symlink_system_library,OPENLIBM,$(LIBMNAME)))
 endif
 
-ifeq ($(APPLE_ARCH),arm64)
+ifeq ($(OS),Darwin)
 $(eval $(call symlink_system_library,CSL,libgcc_s,1.1))
 else
 $(eval $(call symlink_system_library,CSL,libgcc_s,1))
diff --git a/deps/csl.mk b/deps/csl.mk
index e3f84aa..b6dbff3 100644
--- a/deps/csl.mk
+++ b/deps/csl.mk
@@ -76,7 +76,7 @@ else
 $(eval $(call copy_csl,$(call versioned_libname,libgcc_s_seh,1)))
 endif
 else
-ifeq ($(APPLE_ARCH),arm64)
+ifeq ($(OS),Darwin)
 $(eval $(call copy_csl,$(call versioned_libname,libgcc_s,1.1)))
 else
 $(eval $(call copy_csl,$(call versioned_libname,libgcc_s,1)))
