diff -pur SPAdes-3.11.1/src/common/assembly_graph/paths/mapping_path.hpp SPAdes-3.11.1.fixed/src/common/assembly_graph/paths/mapping_path.hpp
--- SPAdes-3.11.1/src/common/assembly_graph/paths/mapping_path.hpp	2017-09-28 11:05:09.000000000 +0200
+++ SPAdes-3.11.1.fixed/src/common/assembly_graph/paths/mapping_path.hpp	2018-05-03 09:47:48.000000000 +0200
@@ -387,7 +387,7 @@ public:
     }
 
     bool has_filling() const {
-        return filling_seq_;
+        return static_cast<bool>(filling_seq_);
     }
 
     Sequence filling_seq() const {
