diff --git a/configure b/configure
index 663c87d..a881b6c 100755
--- a/configure
+++ b/configure
@@ -636,9 +636,6 @@ ac_subst_vars='am__EXEEXT_FALSE
 am__EXEEXT_TRUE
 LTLIBOBJS
 LIBOBJS
-FRIBIDI_ENABLE_DOC_FALSE
-FRIBIDI_ENABLE_DOC_TRUE
-C2MAN
 LDFLAGS_FOR_BUILD
 CPPFLAGS_FOR_BUILD
 CFLAGS_FOR_BUILD
@@ -793,7 +790,6 @@ with_sysroot
 enable_libtool_lock
 enable_debug
 enable_deprecated
-enable_docs
 '
       ac_precious_vars='build_alias
 host_alias
@@ -1439,7 +1435,6 @@ Optional Features:
   --disable-libtool-lock  avoid locking (might break parallel builds)
   --enable-debug          turn on debugging [default=no]
   --disable-deprecated    do not build deprecated functionality
-  --disable-docs          disable building documentation
 
 Optional Packages:
   --with-PACKAGE[=ARG]    use PACKAGE [ARG=yes]
@@ -14609,70 +14604,6 @@ $as_echo "#define FRIBIDI_NO_DEPRECATED 1" >>confdefs.h
 
 fi
 
-# --disable-docs
-# Check whether --enable-docs was given.
-if test "${enable_docs+set}" = set; then :
-  enableval=$enable_docs;
-else
-  enable_docs=yes
-fi
-
-if test x$enable_docs = xyes; then
-	for ac_prog in c2man
-do
-  # Extract the first word of "$ac_prog", so it can be a program name with args.
-set dummy $ac_prog; ac_word=$2
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
-$as_echo_n "checking for $ac_word... " >&6; }
-if ${ac_cv_prog_C2MAN+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-  if test -n "$C2MAN"; then
-  ac_cv_prog_C2MAN="$C2MAN" # Let the user override the test.
-else
-as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
-for as_dir in $PATH
-do
-  IFS=$as_save_IFS
-  test -z "$as_dir" && as_dir=.
-    for ac_exec_ext in '' $ac_executable_extensions; do
-  if as_fn_executable_p "$as_dir/$ac_word$ac_exec_ext"; then
-    ac_cv_prog_C2MAN="$ac_prog"
-    $as_echo "$as_me:${as_lineno-$LINENO}: found $as_dir/$ac_word$ac_exec_ext" >&5
-    break 2
-  fi
-done
-  done
-IFS=$as_save_IFS
-
-fi
-fi
-C2MAN=$ac_cv_prog_C2MAN
-if test -n "$C2MAN"; then
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: $C2MAN" >&5
-$as_echo "$C2MAN" >&6; }
-else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-fi
-
-
-  test -n "$C2MAN" && break
-done
-
-	if test -z "$C2MAN"; then
-		as_fn_error $? "c2man is required to build documentation" "$LINENO" 5
-	fi
-fi
- if test -n "$C2MAN"; then
-  FRIBIDI_ENABLE_DOC_TRUE=
-  FRIBIDI_ENABLE_DOC_FALSE='#'
-else
-  FRIBIDI_ENABLE_DOC_TRUE='#'
-  FRIBIDI_ENABLE_DOC_FALSE=
-fi
-
-
 # Generate output
 ac_config_files="$ac_config_files fribidi.pc lib/fribidi-config.h Makefile gen.tab/Makefile lib/Makefile bin/Makefile doc/Makefile test/Makefile test/unicode-conformance/Makefile"
 
@@ -14825,10 +14756,6 @@ if test -z "${am__fastdepCC_TRUE}" && test -z "${am__fastdepCC_FALSE}"; then
   as_fn_error $? "conditional \"am__fastdepCC\" was never defined.
 Usually this means the macro was only invoked conditionally." "$LINENO" 5
 fi
-if test -z "${FRIBIDI_ENABLE_DOC_TRUE}" && test -z "${FRIBIDI_ENABLE_DOC_FALSE}"; then
-  as_fn_error $? "conditional \"FRIBIDI_ENABLE_DOC\" was never defined.
-Usually this means the macro was only invoked conditionally." "$LINENO" 5
-fi
 
 : "${CONFIG_STATUS=./config.status}"
 ac_write_fail=0
diff --git a/doc/Makefile.in b/doc/Makefile.in
index e908d99..edcdf9f 100644
--- a/doc/Makefile.in
+++ b/doc/Makefile.in
@@ -86,12 +86,6 @@ PRE_UNINSTALL = :
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-@FRIBIDI_ENABLE_DOC_TRUE@am__append_1 = $(inst_symbols:=.3)
-@FRIBIDI_ENABLE_DOC_TRUE@am__append_2 = $(noinst_symbols:=.3)
-@FRIBIDI_ENABLE_DOC_TRUE@am__append_3 = c2man.stamp
-@FRIBIDI_ENABLE_DOC_TRUE@am__append_4 = c2man.stamp.tmp
-@FRIBIDI_ENABLE_DOC_TRUE@am__append_5 = $(man3) c2man.stamp
-@FRIBIDI_ENABLE_DOC_TRUE@am__append_6 = $(includevpath)
 subdir = doc
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/m4/libtool.m4 \
@@ -160,7 +154,7 @@ am__tagged_files = $(HEADERS) $(SOURCES) $(TAGS_FILES) $(LISP)
 am__DIST_COMMON = $(dist_man_MANS) $(srcdir)/Makefile.in \
 	$(top_srcdir)/lib/Headers.mk
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-VPATH = @srcdir@ $(am__append_6)
+VPATH = @srcdir@ $(includevpath)
 ACLOCAL = @ACLOCAL@
 AMTAR = @AMTAR@
 AM_DEFAULT_VERBOSITY = @AM_DEFAULT_VERBOSITY@
@@ -171,7 +165,6 @@ AUTOMAKE = @AUTOMAKE@
 AWK = @AWK@
 BUILD_EXEEXT = @BUILD_EXEEXT@
 BUILD_OBJEXT = @BUILD_OBJEXT@
-C2MAN = @C2MAN@
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
@@ -299,54 +292,52 @@ target_alias = @target_alias@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-dist_man_MANS = $(am__append_1)
-dist_noinst_MANS = $(am__append_2)
-EXTRA_DIST = $(am__append_3)
-MOSTLYCLEANFILES = $(am__append_4)
-MAINTAINERCLEANFILES = $(am__append_5)
+dist_man_MANS = $(inst_symbols:=.3)
+dist_noinst_MANS = $(noinst_symbols:=.3)
+EXTRA_DIST = c2man.stamp
+MOSTLYCLEANFILES = c2man.stamp.tmp
+MAINTAINERCLEANFILES = $(man3) c2man.stamp
 
 # man3
-@FRIBIDI_ENABLE_DOC_TRUE@includepath = -I$(top_builddir)/lib \
-@FRIBIDI_ENABLE_DOC_TRUE@	-I$(top_srcdir)/lib
-@FRIBIDI_ENABLE_DOC_TRUE@includevpath = :$(top_builddir)/lib:$(top_srcdir)/lib
+includepath = -I$(top_builddir)/lib -I$(top_srcdir)/lib
+includevpath = :$(top_builddir)/lib:$(top_srcdir)/lib
 
 # The lib/Headers.mk stuff should come after all optional stuff.
-@FRIBIDI_ENABLE_DOC_TRUE@headers = $(libfribidi_la_headers)
+headers = $(libfribidi_la_headers)
 # The las two lines are not functions, we don't want them here.
-@FRIBIDI_ENABLE_DOC_TRUE@inst_symbols = $(shell sed '$$d' \
-@FRIBIDI_ENABLE_DOC_TRUE@	$(top_srcdir)/lib/fribidi.def | sed \
-@FRIBIDI_ENABLE_DOC_TRUE@	'$$d')
-@FRIBIDI_ENABLE_DOC_TRUE@noinst_symbols = 
-@FRIBIDI_ENABLE_DOC_TRUE@libfribidi_la_headers = \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-arabic.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-begindecls.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-bidi.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-bidi-types.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-bidi-types-list.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-common.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-char-sets.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-char-sets-list.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-deprecated.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-enddecls.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-flags.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-joining.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-joining-types.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-joining-types-list.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-mirroring.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-brackets.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-shape.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-types.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-unicode.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi-unicode-version.h \
-@FRIBIDI_ENABLE_DOC_TRUE@		fribidi.h
+inst_symbols = $(shell sed '$$d' $(top_srcdir)/lib/fribidi.def | sed \
+	'$$d')
+noinst_symbols = 
+libfribidi_la_headers = \
+		fribidi-arabic.h \
+		fribidi-begindecls.h \
+		fribidi-bidi.h \
+		fribidi-bidi-types.h \
+		fribidi-bidi-types-list.h \
+		fribidi-common.h \
+		fribidi-char-sets.h \
+		fribidi-char-sets-list.h \
+		fribidi-deprecated.h \
+		fribidi-enddecls.h \
+		fribidi-flags.h \
+		fribidi-joining.h \
+		fribidi-joining-types.h \
+		fribidi-joining-types-list.h \
+		fribidi-mirroring.h \
+		fribidi-brackets.h \
+		fribidi-shape.h \
+		fribidi-types.h \
+		fribidi-unicode.h \
+		fribidi-unicode-version.h \
+		fribidi.h
 
 # fribidi.h should be the last entry in the list above.
-@FRIBIDI_ENABLE_DOC_TRUE@libfribidi_la_symbols = $(shell cat $(top_srcdir)/lib/fribidi.def)
-@FRIBIDI_ENABLE_DOC_TRUE@man3 = $(dist_man_MANS) $(dist_noinst_MANS)
-@FRIBIDI_ENABLE_DOC_TRUE@C2MANFLAGS = $(includepath) \
-@FRIBIDI_ENABLE_DOC_TRUE@		-D__FRIBIDI_DOC \
-@FRIBIDI_ENABLE_DOC_TRUE@		-DDONT_HAVE_FRIBIDI_CONFIG_H \
-@FRIBIDI_ENABLE_DOC_TRUE@		-M "Programmer's Manual" 
+libfribidi_la_symbols = $(shell cat $(top_srcdir)/lib/fribidi.def)
+man3 = $(dist_man_MANS) $(dist_noinst_MANS)
+C2MANFLAGS = $(includepath) \
+		-D__FRIBIDI_DOC \
+		-DDONT_HAVE_FRIBIDI_CONFIG_H \
+		-M "Programmer's Manual" 
 
 all: all-am
 
@@ -592,21 +583,22 @@ uninstall-man: uninstall-man3
 .PRECIOUS: Makefile
 
 
-@FRIBIDI_ENABLE_DOC_TRUE@c2man.stamp: $(headers)
-@FRIBIDI_ENABLE_DOC_TRUE@	@$(RM) $@.tmp
-@FRIBIDI_ENABLE_DOC_TRUE@	@touch $@.tmp
-@FRIBIDI_ENABLE_DOC_TRUE@	@($(LN_S) $^ . || cp $^ .)
-@FRIBIDI_ENABLE_DOC_TRUE@	@echo Running c2man
-@FRIBIDI_ENABLE_DOC_TRUE@	@$(C2MAN) -T n $(C2MANFLAGS) $(headers) 2>&1
-@FRIBIDI_ENABLE_DOC_TRUE@	@$(RM) $(headers)
-@FRIBIDI_ENABLE_DOC_TRUE@	@mv $@.tmp $@
-
-@FRIBIDI_ENABLE_DOC_TRUE@$(man3): c2man.stamp
-@FRIBIDI_ENABLE_DOC_TRUE@	@if ! test -f $@; then \
-@FRIBIDI_ENABLE_DOC_TRUE@		$(RM) $<; \
-@FRIBIDI_ENABLE_DOC_TRUE@		$(MAKE) $(AM_MAKEFLAGS) $<;\
-@FRIBIDI_ENABLE_DOC_TRUE@	fi
-@FRIBIDI_ENABLE_DOC_TRUE@	@touch $@
+c2man.stamp: $(headers)
+	@$(RM) $@.tmp
+	@touch $@.tmp
+	@($(LN_S) $^ . || cp $^ .)
+	@echo Running c2man
+	@$(top_srcdir)/missing --run c2man -T n $(C2MANFLAGS) $(headers) 2>&1 \
+	|| true
+	@$(RM) $(headers)
+	@mv $@.tmp $@
+
+$(man3): c2man.stamp
+	@if ! test -f $@; then \
+		$(RM) $<; \
+		$(MAKE) $(AM_MAKEFLAGS) $<;\
+	fi
+	@touch $@
 
 -include $(top_srcdir)/git.mk
 
