diff --git a/Modules/ThirdParty/MINC/src/libminc/CMakeLists.txt b/Modules/ThirdParty/MINC/src/libminc/CMakeLists.txt
index b252659..c418895 100644
--- a/Modules/ThirdParty/MINC/src/libminc/CMakeLists.txt
+++ b/Modules/ThirdParty/MINC/src/libminc/CMakeLists.txt
@@ -538,7 +538,7 @@ IF(LIBMINC_INSTALL_LIB_DIR AND NOT LIBMINC_INSTALL_NO_DEVELOPMENT)
      ${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/Use${LIBMINC_EXTERNAL_LIB_PREFIX}LIBMINC.cmake 
      ${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/${LIBMINC_EXTERNAL_LIB_PREFIX}LIBMINCConfig.cmake
     DESTINATION 
-     ${LIBMINC_INSTALL_LIB_DIR}
+     ${LIBMINC_INSTALL_LIB_DIR}/cmake
     COMPONENT Development)
   
 ENDIF(LIBMINC_INSTALL_LIB_DIR AND NOT LIBMINC_INSTALL_NO_DEVELOPMENT)
