diff -pur upscaledb-2.2.0/src/2aes/aes.h upscaledb-2.2.0-fixed/src/2aes/aes.h
--- upscaledb-2.2.0/src/2aes/aes.h	2016-04-03 21:14:50.000000000 +0200
+++ upscaledb-2.2.0-fixed/src/2aes/aes.h	2019-09-07 18:01:05.000000000 +0200
@@ -48,19 +48,19 @@ class AesCipher {
     AesCipher(const uint8_t key[kAesBlockSize], uint64_t salt = 0) {
       uint64_t iv[2] = {salt, 0};
 
-	    EVP_CIPHER_CTX_init(&m_encrypt_ctx);
-	    EVP_EncryptInit_ex(&m_encrypt_ctx, EVP_aes_128_cbc(), NULL, key,
+	    m_encrypt_ctx = EVP_CIPHER_CTX_new();
+	    EVP_EncryptInit_ex(m_encrypt_ctx, EVP_aes_128_cbc(), NULL, key,
 			  			(uint8_t *)&iv[0]);
-	    EVP_CIPHER_CTX_init(&m_decrypt_ctx);
-	    EVP_DecryptInit_ex(&m_decrypt_ctx, EVP_aes_128_cbc(), NULL, key,
+	    m_decrypt_ctx = EVP_CIPHER_CTX_new();
+	    EVP_DecryptInit_ex(m_decrypt_ctx, EVP_aes_128_cbc(), NULL, key,
 			  			(uint8_t *)&iv[0]);
-	    EVP_CIPHER_CTX_set_padding(&m_encrypt_ctx, 0);
-	    EVP_CIPHER_CTX_set_padding(&m_decrypt_ctx, 0);
+	    EVP_CIPHER_CTX_set_padding(m_encrypt_ctx, 0);
+	    EVP_CIPHER_CTX_set_padding(m_decrypt_ctx, 0);
     }
 
     ~AesCipher() {
-      EVP_CIPHER_CTX_cleanup(&m_encrypt_ctx);
-  	  EVP_CIPHER_CTX_cleanup(&m_decrypt_ctx);
+      EVP_CIPHER_CTX_free(m_encrypt_ctx);
+  	  EVP_CIPHER_CTX_free(m_decrypt_ctx);
     }
 
     /*
@@ -75,11 +75,11 @@ class AesCipher {
 	    /* update ciphertext, c_len is filled with the length of ciphertext
 	     * generated, len is the size of plaintext in bytes */
 	    int clen = (int)len;
-	    EVP_EncryptUpdate(&m_encrypt_ctx, ciphertext, &clen, plaintext, (int)len);
+	    EVP_EncryptUpdate(m_encrypt_ctx, ciphertext, &clen, plaintext, (int)len);
 
 	    /* update ciphertext with the final remaining bytes */
 	    int outlen;
-	    EVP_EncryptFinal(&m_encrypt_ctx, ciphertext + clen, &outlen);
+	    EVP_EncryptFinal(m_encrypt_ctx, ciphertext + clen, &outlen);
     }
 
     /*
@@ -92,13 +92,13 @@ class AesCipher {
       assert(len % kAesBlockSize == 0);
 
 	    int plen = (int)len, flen = 0;
-	    EVP_DecryptUpdate(&m_decrypt_ctx, plaintext, &plen, ciphertext, (int)len);
-	    EVP_DecryptFinal(&m_decrypt_ctx, plaintext + plen, &flen);
+	    EVP_DecryptUpdate(m_decrypt_ctx, plaintext, &plen, ciphertext, (int)len);
+	    EVP_DecryptFinal(m_decrypt_ctx, plaintext + plen, &flen);
     }
 
   private:
-    EVP_CIPHER_CTX m_encrypt_ctx;
-    EVP_CIPHER_CTX m_decrypt_ctx;
+    EVP_CIPHER_CTX *m_encrypt_ctx;
+    EVP_CIPHER_CTX *m_decrypt_ctx;
 };
 
 } // namespace upscaledb
