diff --git a/lib/Bi/Builder.pm b/lib/Bi/Builder.pm
index e60e8b61..68d2f19c 100644
--- a/lib/Bi/Builder.pm
+++ b/lib/Bi/Builder.pm
@@ -59,7 +59,7 @@ Enable diagnostic outputs to standard error.
 
 Use single-precision floating point.
 
-=item C<--enable-openmp> (default on)
+=item C<--enable-openmp> (default off)
 
 Use OpenMP multithreading.
 
@@ -139,7 +139,7 @@ sub new {
         _force => 0,
         _warnings => 0,
         _assert => 1,
-        _openmp => 1,
+        _openmp => 0,
         _cuda => 0,
         _gpu_cache => 0,
         _sse => 0,
