diff --git a/em++ b/em++
index 06f3428..20c1240 100755
--- a/em++
+++ b/em++
@@ -18,4 +18,4 @@ if sys.version_info.major == 2:
 else:
   import os, subprocess
   if __name__ == '__main__':
-    sys.exit(subprocess.call(['python2', os.path.join(os.path.dirname(__file__), 'emcc.py')] + sys.argv[1:]))
+    sys.exit(subprocess.call(['python2', os.path.join(os.path.dirname(os.path.realpath(__file__)), 'emcc.py')] + sys.argv[1:]))
diff --git a/em++.py b/em++.py
index 85236eb..0839d8d 100644
--- a/em++.py
+++ b/em++.py
@@ -14,4 +14,4 @@ if sys.version_info.major == 2:
 else:
   import os, subprocess
   if __name__ == '__main__':
-    sys.exit(subprocess.call(['python2', os.path.join(os.path.dirname(__file__), 'emcc.py')] + sys.argv[1:]))
+    sys.exit(subprocess.call(['python2', os.path.join(os.path.dirname(os.path.realpath(__file__)), 'emcc.py')] + sys.argv[1:]))
diff --git a/emar b/emar
index 3eefa59..f8ac832 100755
--- a/emar
+++ b/emar
@@ -15,4 +15,4 @@ if sys.version_info.major == 2:
 else:
   import os, subprocess
   if __name__ == '__main__':
-    sys.exit(subprocess.call(['python2', os.path.join(os.path.dirname(__file__), 'emar.py')] + sys.argv[1:]))
+    sys.exit(subprocess.call(['python2', os.path.join(os.path.dirname(os.path.realpath(__file__)), 'emar.py')] + sys.argv[1:]))
diff --git a/emcc b/emcc
index 7f7af40..a5042da 100755
--- a/emcc
+++ b/emcc
@@ -15,4 +15,4 @@ if sys.version_info.major == 2:
 else:
   import os, subprocess
   if __name__ == '__main__':
-    sys.exit(subprocess.call(['python2', os.path.join(os.path.dirname(__file__), 'emcc.py')] + sys.argv[1:]))
+    sys.exit(subprocess.call(['python2', os.path.join(os.path.dirname(os.path.realpath(__file__)), 'emcc.py')] + sys.argv[1:]))
diff --git a/emcmake b/emcmake
index f4cead7..a8b280e 100755
--- a/emcmake
+++ b/emcmake
@@ -14,4 +14,4 @@ if sys.version_info.major == 2:
 else:
   import os, subprocess
   if __name__ == '__main__':
-    sys.exit(subprocess.call(['python2', os.path.join(os.path.dirname(__file__), 'emcmake.py')] + sys.argv[1:]))
+    sys.exit(subprocess.call(['python2', os.path.join(os.path.dirname(os.path.realpath(__file__)), 'emcmake.py')] + sys.argv[1:]))
diff --git a/emconfigure b/emconfigure
index 35d3d70..aec46cf 100755
--- a/emconfigure
+++ b/emconfigure
@@ -14,4 +14,4 @@ if sys.version_info.major == 2:
 else:
   import os, subprocess
   if __name__ == '__main__':
-    sys.exit(subprocess.call(['python2', os.path.join(os.path.dirname(__file__), 'emconfigure.py')] + sys.argv[1:]))
+    sys.exit(subprocess.call(['python2', os.path.join(os.path.dirname(os.path.realpath(__file__)), 'emconfigure.py')] + sys.argv[1:]))
diff --git a/emmake b/emmake
index 83ea071..ad13b6c 100755
--- a/emmake
+++ b/emmake
@@ -17,4 +17,4 @@ if sys.version_info.major == 2:
 else:
   import os, subprocess
   if __name__ == '__main__':
-    sys.exit(subprocess.call(['python2', os.path.join(os.path.dirname(__file__), 'emmake.py')] + sys.argv[1:]))
+    sys.exit(subprocess.call(['python2', os.path.join(os.path.dirname(os.path.realpath(__file__)), 'emmake.py')] + sys.argv[1:]))
