diff -pur a/aclocal.m4 b/aclocal.m4
--- a/aclocal.m4	2018-12-05 15:31:34.000000000 +0100
+++ b/aclocal.m4	2020-11-08 11:05:14.000000000 +0100
@@ -300,7 +300,8 @@ AC_DEFUN(BASH_FUNC_STRSIGNAL,
 [AC_MSG_CHECKING([for the existence of strsignal])
 AC_CACHE_VAL(bash_cv_have_strsignal,
 [AC_TRY_LINK([#include <sys/types.h>
-#include <signal.h>],
+#include <signal.h>
+#include <string.h>],
 [char *s = (char *)strsignal(2);],
  bash_cv_have_strsignal=yes, bash_cv_have_strsignal=no)])
 AC_MSG_RESULT($bash_cv_have_strsignal)
@@ -4068,6 +4069,7 @@ AC_DEFUN([BASH_FUNC_SNPRINTF],
     AC_CACHE_CHECK([for standard-conformant snprintf], [bash_cv_func_snprintf],
       [AC_TRY_RUN([
 #include <stdio.h>
+#include <stdlib.h>
 
 main()
 {
diff -pur a/configure b/configure
--- a/configure	2019-01-02 15:43:31.000000000 +0100
+++ b/configure	2020-11-08 11:06:16.000000000 +0100
@@ -14568,6 +14568,7 @@ else
 /* end confdefs.h.  */
 #include <sys/types.h>
 #include <signal.h>
+#include <string.h>
 int
 main ()
 {
@@ -14999,6 +15000,7 @@ else
 /* end confdefs.h.  */
 
 #include <stdio.h>
+#include <stdlib.h>
 
 main()
 {
