diff -pur mono-6.12.0.182/mono/utils/mono-threads-mach-helper.c mono-6.12.0.182-patched/mono/utils/mono-threads-mach-helper.c
--- mono-6.12.0.182/mono/utils/mono-threads-mach-helper.c	2022-06-14 22:36:03
+++ mono-6.12.0.182-patched/mono/utils/mono-threads-mach-helper.c	2022-12-01 10:37:05
@@ -59,11 +59,7 @@ mono_dead_letter_dealloc (id self, SEL _cmd)
 {
 	struct objc_super super;
 	super.receiver = self;
-#if !defined(__cplusplus) && !__OBJC2__
-	super.class = nsobject;
-#else
 	super.super_class = nsobject;
-#endif
 	void (*objc_msgSendSuper_op)(struct objc_super *, SEL) = (void (*)(struct objc_super *, SEL)) objc_msgSendSuper;
 	objc_msgSendSuper_op (&super, dealloc);
 
