diff --git a/imagewin/scale_xbr.h b/imagewin/scale_xbr.h
index c813cc0240b80fe49c47ca0316ed3be84ebe1eb0..0d9843e2a47e08f2de6840ad7b561b544d0033c8 100644
--- a/imagewin/scale_xbr.h
+++ b/imagewin/scale_xbr.h
@@ -99,14 +99,14 @@ struct RGBColor {
 	}
 	// See http://www.compuphase.com/cmetric.htm
 	int dist(RGBColor<Manip_pixels, tol> const &other) const {
-		return abs(dr - other.dr) +
-		       abs(dg - other.dg) +
-		       abs(db - other.db);
+		return abs(static_cast<int>(dr - other.dr)) +
+		       abs(static_cast<int>(dg - other.dg)) +
+		       abs(static_cast<int>(db - other.db));
 	}
 	bool equals(RGBColor<Manip_pixels, tol> const &other) const {
-		return abs(dr - other.dr) <= (tol *  9)
-		       && abs(dg - other.dg) <= (tol * 16)
-		       && abs(db - other.db) <= (tol *  4);
+		return abs(static_cast<int>(dr - other.dr)) <= (tol *  9)
+		       && abs(static_cast<int>(dg - other.dg)) <= (tol * 16)
+		       && abs(static_cast<int>(db - other.db)) <= (tol *  4);
 	}
 	template <unsigned int N, unsigned int M>
 	inline void blend(RGBColor<Manip_pixels, tol> const &other) {
diff --git a/usecode/ucinternal.cc b/usecode/ucinternal.cc
index 8bdf984cf138122ff92dbd46c8e77f5b816b91cb..f0346ea0dcf6e0faf95d2ff605aa6ff7f7a3e7ec 100644
--- a/usecode/ucinternal.cc
+++ b/usecode/ucinternal.cc
@@ -638,7 +638,7 @@ int Usecode_internal::get_face_shape(
 	npc = as_actor(get_item(arg1));
 	int shape = -1;
 	if (arg1.is_int()) {
-		shape = abs(arg1.get_int_value());
+		shape = std::abs(arg1.get_int_value());
 		if (shape == 356)   // Avatar.
 			shape = 0;
 	} else if (npc)
